{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "from collections import defaultdict\n",
    "import ftplib\n",
    "import getpass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#import urllib3\n",
    "#urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "# explain above, and warn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n",
    "from bioblend.galaxy import GalaxyInstance\n",
    "from bioblend.galaxy.histories import HistoryClient\n",
    "from bioblend.galaxy.datasets import DatasetClient"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "#Need to create a file with this\n",
    "with open('../galaxy.api') as f:\n",
    "    api_key = f.readline().rstrip()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "server = 'https://bioinformatics.dbs.umt.edu'\n",
    "ftp_server = 'bioinformatics.dbs.umt.edu'\n",
    "ftp_user = 'tiago.antao@mso.umt.edu'\n",
    "ftp_port = 8021\n",
    "history_name = 'data_science_example'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "········\n"
     ]
    }
   ],
   "source": [
    "ftp_pass = getpass.getpass()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "gi = GalaxyInstance(url=server, key=api_key)\n",
    "gi.verify = False\n",
    "histories = gi.histories"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "data_science_example\n",
      "Jenya\n",
      "Unnamed history\n",
      "Marc\n",
      "brian\n",
      "Data Manager History (automatically created)\n",
      "reference genomes\n",
      "Unnamed history\n"
     ]
    }
   ],
   "source": [
    "for history in histories.get_histories():\n",
    "    if history['name'] == history_name:\n",
    "        histories.delete_history(history['id'])\n",
    "    print(history['name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "ds_history = histories.create_history(history_name)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "ftp = ftplib.FTP() \n",
    "ftp.connect(host=ftp_server,port=ftp_port)\n",
    "ftp.login(user=ftp_user, passwd=ftp_pass)\n",
    "f = open('../raw/part-3L.vcf.gz','rb')\n",
    "ftp.set_pasv(False)  # explain\n",
    "ftp.storbinary('STOR part-3L.vcf.gz', f)\n",
    "f.close() \n",
    "ftp.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'implicit_collections': [],\n",
       " 'jobs': [{'create_time': '2016-08-10T17:12:39.076756',\n",
       "   'exit_code': None,\n",
       "   'id': '0e74a7c1bf7ec99d',\n",
       "   'model_class': 'Job',\n",
       "   'state': 'new',\n",
       "   'tool_id': 'upload1',\n",
       "   'update_time': '2016-08-10T17:12:39.120913'}],\n",
       " 'output_collections': [],\n",
       " 'outputs': [{'create_time': '2016-08-10T17:12:38.982612',\n",
       "   'data_type': 'galaxy.datatypes.data.Text',\n",
       "   'deleted': False,\n",
       "   'file_ext': 'auto',\n",
       "   'file_size': 0,\n",
       "   'genome_build': '?',\n",
       "   'hda_ldda': 'hda',\n",
       "   'hid': 1,\n",
       "   'history_content_type': 'dataset',\n",
       "   'history_id': '7d83dea28f2ec7cc',\n",
       "   'id': '56573e25b995be9d',\n",
       "   'metadata_data_lines': None,\n",
       "   'metadata_dbkey': '?',\n",
       "   'misc_blurb': None,\n",
       "   'misc_info': None,\n",
       "   'model_class': 'HistoryDatasetAssociation',\n",
       "   'name': 'part-3L.vcf.gz',\n",
       "   'output_name': 'output0',\n",
       "   'peek': '<table cellspacing=\"0\" cellpadding=\"3\"></table>',\n",
       "   'purged': False,\n",
       "   'state': 'queued',\n",
       "   'tags': [],\n",
       "   'update_time': '2016-08-10T17:12:39.020363',\n",
       "   'uuid': '6aa7d4ae-19b6-4afb-b9c9-669c1de3a78e',\n",
       "   'visible': True}]}"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gi.tools.upload_from_ftp('part-3L.vcf.gz', ds_history['id'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'dataset_id': 'a6f7719d361bdab3',\n",
       "  'deleted': False,\n",
       "  'extension': 'auto',\n",
       "  'hid': 1,\n",
       "  'history_content_type': 'dataset',\n",
       "  'history_id': '7d83dea28f2ec7cc',\n",
       "  'id': '56573e25b995be9d',\n",
       "  'name': 'part-3L.vcf.gz',\n",
       "  'purged': False,\n",
       "  'state': 'running',\n",
       "  'type': 'file',\n",
       "  'type_id': 'dataset-56573e25b995be9d',\n",
       "  'url': '/api/histories/7d83dea28f2ec7cc/contents/56573e25b995be9d',\n",
       "  'visible': True}]"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "contents = gi.histories.show_history(ds_history['id'], contents=True)\n",
    "contents"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def summarize_contents(contents):\n",
    "    summary = defaultdict(list)\n",
    "    for item in contents:\n",
    "        summary['íd'].append(item['id'])\n",
    "        summary['híd'].append(item['hid'])\n",
    "        summary['name'].append(item['name'])\n",
    "        summary['type'].append(item['type'])\n",
    "        summary['extension'].append(item['extension'])\n",
    "    return pd.DataFrame.from_dict(summary)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>extension</th>\n",
       "      <th>híd</th>\n",
       "      <th>name</th>\n",
       "      <th>type</th>\n",
       "      <th>íd</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>auto</td>\n",
       "      <td>1</td>\n",
       "      <td>part-3L.vcf.gz</td>\n",
       "      <td>file</td>\n",
       "      <td>56573e25b995be9d</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  extension  híd            name  type                íd\n",
       "0      auto    1  part-3L.vcf.gz  file  56573e25b995be9d"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "summarize_contents(contents)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'dataset_id': 'a6f7719d361bdab3',\n",
       " 'deleted': False,\n",
       " 'extension': 'auto',\n",
       " 'hid': 1,\n",
       " 'history_content_type': 'dataset',\n",
       " 'history_id': '7d83dea28f2ec7cc',\n",
       " 'id': '56573e25b995be9d',\n",
       " 'name': 'part-3L.vcf.gz',\n",
       " 'purged': False,\n",
       " 'state': 'running',\n",
       " 'type': 'file',\n",
       " 'type_id': 'dataset-56573e25b995be9d',\n",
       " 'url': '/api/histories/7d83dea28f2ec7cc/contents/56573e25b995be9d',\n",
       " 'visible': True}"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf_ds = contents[0]\n",
    "vcf_ds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "collapsed": false,
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[{'config_file': '/galaxy-central/tools/data_source/ucsc_tablebrowser.xml',\n",
       "  'description': 'table browser',\n",
       "  'id': 'ucsc_table_direct1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=ucsc_table_direct1',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'UCSC Main',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/maf/maf_to_bed.xml',\n",
       "  'description': 'Converts a MAF formatted file to the BED format',\n",
       "  'id': 'MAF_To_BED1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=MAF_To_BED1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MAF to BED',\n",
       "  'panel_section_id': 'convert',\n",
       "  'panel_section_name': 'Convert Formats',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/table_annovar/08b003ee9db7/table_annovar/table_annovar.xml',\n",
       "  'description': 'with functional information using ANNOVAR',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/table_annovar/table_annovar/0.2',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Ftable_annovar%2Ftable_annovar%2F0.2',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'ANNOVAR Annotate VCF',\n",
       "  'panel_section_id': 'ngs:_variant_analysis',\n",
       "  'panel_section_name': 'NGS: Variant Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '08b003ee9db7',\n",
       "   'name': 'table_annovar',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.2'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_CollectAlignmentSummaryMetrics.xml',\n",
       "  'description': 'writes a file containing summary alignment metrics',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_CASM/1.136.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_CASM%2F1.136.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Collect Alignment Summary Metrics',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_infoseq.xml',\n",
       "  'description': 'Displays some simple information about sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: infoseq46/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+infoseq46%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'infoseq',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/7f3c448e119b/ncbi_blast_plus/tools/ncbi_blast_plus/ncbi_rpstblastn_wrapper.xml',\n",
       "  'description': 'Search protein domain database (PSSMs) with translated nucleotide query sequence(s)',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_rpstblastn_wrapper/0.1.07',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fncbi_blast_plus%2Fncbi_rpstblastn_wrapper%2F0.1.07',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'NCBI BLAST+ rpstblastn',\n",
       "  'panel_section_id': 'ngs:_mapping',\n",
       "  'panel_section_name': 'NGS: Mapping',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '7f3c448e119b',\n",
       "   'name': 'ncbi_blast_plus',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.1.07'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/fasta_filter_by_length/c8cd0a03db49/fasta_filter_by_length/fasta_filter_by_length.xml',\n",
       "  'description': '',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/fasta_filter_by_length/fasta_filter_by_length/1.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Ffasta_filter_by_length%2Ffasta_filter_by_length%2F1.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Filter sequences by length',\n",
       "  'panel_section_id': 'ngs:_qc_and_manipulation',\n",
       "  'panel_section_name': 'NGS: QC and manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'c8cd0a03db49',\n",
       "   'name': 'fasta_filter_by_length',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/288462ec2630/text_processing/tac.xml',\n",
       "  'description': 'reverse a file (reverse cat)',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_tac/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Ftext_processing%2Ftp_tac%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'tac',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '288462ec2630',\n",
       "   'name': 'text_processing',\n",
       "   'owner': 'bgruening',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/metabolicmine.xml',\n",
       "  'description': 'server',\n",
       "  'id': 'metabolicmine',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=metabolicmine',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'metabolicMine',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/maf_to_interval_converter.xml',\n",
       "  'description': '',\n",
       "  'id': 'CONVERTER_maf_to_interval_0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=CONVERTER_maf_to_interval_0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert MAF to Genomic Intervals',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.2'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/koen/gbsx/3b325560887b/gbsx/gbsx_demultiplexer.xml',\n",
       "  'description': 'demultiplexes inline barcodes from GBS and RAD experiments',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/koen/gbsx/gbsx_demultiplexer/1.1.5',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fkoen%2Fgbsx%2Fgbsx_demultiplexer%2F1.1.5',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'GBSX: Demultiplexer',\n",
       "  'panel_section_id': 'ngs:_rads',\n",
       "  'panel_section_name': 'NGS: RADs',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '3b325560887b',\n",
       "   'name': 'gbsx',\n",
       "   'owner': 'koen',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.1.5'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_helixturnhelix.xml',\n",
       "  'description': 'Report nucleic acid binding motifs',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: helixturnhelix43/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+helixturnhelix43%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'helixturnhelix',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/interval_to_bed12_converter.xml',\n",
       "  'description': '',\n",
       "  'id': 'CONVERTER_interval_to_bed12_0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=CONVERTER_interval_to_bed12_0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert Genomic Intervals To Strict BED12',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/tagBed.xml',\n",
       "  'description': 'tag BAM alignments based on overlaps with interval files',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_tagbed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_tagbed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'TagBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/lped_to_fped_converter.xml',\n",
       "  'description': '',\n",
       "  'id': 'lped2fpedconvert',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=lped2fpedconvert',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert lped to fped',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '0.01'},\n",
       " {'config_file': '/galaxy-central/tools/filters/gff/gff_filter_by_attribute.xml',\n",
       "  'description': 'using simple expressions',\n",
       "  'id': 'gff_filter_by_attribute',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=gff_filter_by_attribute',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Filter GFF data by attribute',\n",
       "  'panel_section_id': 'filter',\n",
       "  'panel_section_name': 'Filter and Sort',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/nick/sequence_content_trimmer/7f170cb06e2e/sequence_content_trimmer/trimmer.xml',\n",
       "  'description': 'trim reads based on certain bases',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/nick/sequence_content_trimmer/sequence_content_trimmer/0.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fnick%2Fsequence_content_trimmer%2Fsequence_content_trimmer%2F0.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Sequence Content Trimmer',\n",
       "  'panel_section_id': 'ngs:_de_novo',\n",
       "  'panel_section_name': 'NGS: De Novo',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '7f170cb06e2e',\n",
       "   'name': 'sequence_content_trimmer',\n",
       "   'owner': 'nick',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/data_manager_bwa_mem_index_builder/cb0147ade868/data_manager_bwa_mem_index_builder/data_manager/bwa_mem_index_builder.xml',\n",
       "  'description': 'builder',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/data_manager_bwa_mem_index_builder/bwa_mem_index_builder_data_manager/0.0.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fdata_manager_bwa_mem_index_builder%2Fbwa_mem_index_builder_data_manager%2F0.0.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'DataManagerTool',\n",
       "  'name': 'BWA-MEM index',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'cb0147ade868',\n",
       "   'name': 'data_manager_bwa_mem_index_builder',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.0.1'},\n",
       " {'config_file': '/galaxy-central/tools/filters/wiggle_to_simple.xml',\n",
       "  'description': 'converter',\n",
       "  'id': 'wiggle2simple1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=wiggle2simple1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Wiggle-to-Interval',\n",
       "  'panel_section_id': 'convert',\n",
       "  'panel_section_name': 'Convert Formats',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_CollectInsertSizeMetrics.xml',\n",
       "  'description': 'plots distribution of insert sizes',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_CollectInsertSizeMetrics/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_CollectInsertSizeMetrics%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'CollectInsertSizeMetrics',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/macs2/6d4babad010f/macs2/macs2_refinepeak.xml',\n",
       "  'description': 'Refine peak summits and give scores measuring balance of forward- backward tags (Experimental)',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_refinepeak/2.1.0.20151222.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fmacs2%2Fmacs2_refinepeak%2F2.1.0.20151222.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MACS2 refinepeak',\n",
       "  'panel_section_id': 'ngs:_peak_calling',\n",
       "  'panel_section_name': 'NGS: Peak Calling',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '6d4babad010f',\n",
       "   'name': 'macs2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.1.0.20151222.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/288462ec2630/text_processing/easyjoin.xml',\n",
       "  'description': 'two files',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_easyjoin_tool/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Ftext_processing%2Ftp_easyjoin_tool%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Join',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '288462ec2630',\n",
       "   'name': 'text_processing',\n",
       "   'owner': 'bgruening',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_msbar.xml',\n",
       "  'description': 'Mutate sequence beyond all recognition',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: msbar55/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+msbar55%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'msbar',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_pepinfo.xml',\n",
       "  'description': 'Plots simple amino acid properties in parallel',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: pepinfo69/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+pepinfo69%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'pepinfo',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/gatk2/84584664264c/gatk2/variant_combine.xml',\n",
       "  'description': '',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/gatk2/gatk2_variant_combine/2.8.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fgatk2%2Fgatk2_variant_combine%2F2.8.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Combine Variants',\n",
       "  'panel_section_id': 'ngs:_gatk',\n",
       "  'panel_section_name': 'NGS: GATK',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '84584664264c',\n",
       "   'name': 'gatk2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.8.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_digest.xml',\n",
       "  'description': 'Protein proteolytic enzyme or reagent cleavage digest',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: digest23/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+digest23%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'digest',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/tailWrapper.xml',\n",
       "  'description': 'lines from a dataset',\n",
       "  'id': 'Show tail1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=Show+tail1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Select last',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_extractfeat.xml',\n",
       "  'description': 'Extract features from a sequence',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: extractfeat34/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+extractfeat34%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'extractfeat',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_ReplaceSamHeader.xml',\n",
       "  'description': 'replace header in a SAM/BAM dataset',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_ReplaceSamHeader/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_ReplaceSamHeader%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'ReplaceSamHeader',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/portiahollyoak/fastuniq/816cb55b5a2d/fastuniq/fastuniq.xml',\n",
       "  'description': '',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/portiahollyoak/fastuniq/fastuniq/1.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fportiahollyoak%2Ffastuniq%2Ffastuniq%2F1.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'FastUniq',\n",
       "  'panel_section_id': 'ngs:_qc_and_manipulation',\n",
       "  'panel_section_name': 'NGS: QC and manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '816cb55b5a2d',\n",
       "   'name': 'fastuniq',\n",
       "   'owner': 'portiahollyoak',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.1'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/interval_to_bedstrict_converter.xml',\n",
       "  'description': '',\n",
       "  'id': 'CONVERTER_interval_to_bedstrict_0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=CONVERTER_interval_to_bedstrict_0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert Genomic Intervals To Strict BED',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/bedpeToBam.xml',\n",
       "  'description': 'converter',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bedpetobam/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_bedpetobam%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'BEDPE to BAM',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/axt_to_lav.xml',\n",
       "  'description': 'Converts an AXT formatted file to LAV format',\n",
       "  'id': 'axt_to_lav_1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=axt_to_lav_1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'AXT to LAV',\n",
       "  'panel_section_id': 'convert',\n",
       "  'panel_section_name': 'Convert Formats',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/gff_to_fli_converter.xml',\n",
       "  'description': '',\n",
       "  'id': 'CONVERTER_gff_to_fli_0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=CONVERTER_gff_to_fli_0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert GFF to Feature Location Index',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/288462ec2630/text_processing/sort.xml',\n",
       "  'description': 'data in ascending or descending order',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_header_tool/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Ftext_processing%2Ftp_sort_header_tool%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Sort',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '288462ec2630',\n",
       "   'name': 'text_processing',\n",
       "   'owner': 'bgruening',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/48b5f719e36a/str_fm/probvalueforhetero.xml',\n",
       "  'description': '',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/heteroprob/2.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Farkarachai-fungtammasan%2Fstr_fm%2Fheteroprob%2F2.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Evaluate the probability of the allele combination to generate read profile',\n",
       "  'panel_section_id': 'str-fm:_microsatellite_analysis',\n",
       "  'panel_section_name': 'STR-FM: Microsatellite Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '48b5f719e36a',\n",
       "   'name': 'str_fm',\n",
       "   'owner': 'arkarachai-fungtammasan',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.0.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/sam_to_bam.xml',\n",
       "  'description': '',\n",
       "  'id': 'CONVERTER_sam_to_bam',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=CONVERTER_sam_to_bam',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert SAM to BAM',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '2.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/288462ec2630/text_processing/replace_text_in_line.xml',\n",
       "  'description': 'in entire line',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_line/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Ftext_processing%2Ftp_replace_in_line%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Replace Text',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '288462ec2630',\n",
       "   'name': 'text_processing',\n",
       "   'owner': 'bgruening',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_octanol.xml',\n",
       "  'description': 'Displays protein hydropathy',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: octanol63/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+octanol63%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'octanol',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_banana.xml',\n",
       "  'description': 'Bending and curvature plot in B-DNA',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: banana3/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+banana3%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'banana',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/groupbyBed.xml',\n",
       "  'description': 'group by common cols and summarize other cols',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_groupbybed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_groupbybed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'GroupByBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_geecee.xml',\n",
       "  'description': 'Calculates fractional GC content of nucleic acid sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: geecee41/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+geecee41%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'geecee',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/fastq_paired_end_splitter/c80bce242eec/fastq_paired_end_splitter/fastq_paired_end_splitter.xml',\n",
       "  'description': 'on joined paired end reads',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/fastq_paired_end_splitter/fastq_paired_end_splitter/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Ffastq_paired_end_splitter%2Ffastq_paired_end_splitter%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'FASTQ splitter',\n",
       "  'panel_section_id': 'ngs:_qc_and_manipulation',\n",
       "  'panel_section_name': 'NGS: QC and manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'c80bce242eec',\n",
       "   'name': 'fastq_paired_end_splitter',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/288462ec2630/text_processing/unfold_column.xml',\n",
       "  'description': 'columns from a table',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_unfold_column_tool/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Ftext_processing%2Ftp_unfold_column_tool%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Unfold',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '288462ec2630',\n",
       "   'name': 'text_processing',\n",
       "   'owner': 'bgruening',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/data_manager_snpeff/9ac823a8b328/data_manager_snpeff/data_manager/data_manager_snpEff_databases.xml',\n",
       "  'description': 'Read the list of available snpEff databases',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/data_manager_snpeff/data_manager_snpeff_databases/4.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fdata_manager_snpeff%2Fdata_manager_snpeff_databases%2F4.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'DataManagerTool',\n",
       "  'name': 'SnpEff Databases',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '9ac823a8b328',\n",
       "   'name': 'data_manager_snpeff',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '4.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/cufflinks/9aab29e159a7/cufflinks/cufflinks_wrapper.xml',\n",
       "  'description': 'transcript assembly and FPKM (RPKM) estimates for RNA-Seq data',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/cufflinks/cufflinks/0.0.7',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fcufflinks%2Fcufflinks%2F0.0.7',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Cufflinks',\n",
       "  'panel_section_id': 'ngs:_rna_analysis',\n",
       "  'panel_section_name': 'NGS: RNA Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '9aab29e159a7',\n",
       "   'name': 'cufflinks',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.0.7'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/sortBed.xml',\n",
       "  'description': 'order the intervals',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_sortbed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_sortbed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'SortBED',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/48b5f719e36a/str_fm/microsatcompat.xml',\n",
       "  'description': '',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/microsatcompat/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Farkarachai-fungtammasan%2Fstr_fm%2Fmicrosatcompat%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Check STR motif compatibility between reference and read STRs',\n",
       "  'panel_section_id': 'str-fm:_microsatellite_analysis',\n",
       "  'panel_section_name': 'STR-FM: Microsatellite Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '48b5f719e36a',\n",
       "   'name': 'str_fm',\n",
       "   'owner': 'arkarachai-fungtammasan',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/bed_to_fli_converter.xml',\n",
       "  'description': '',\n",
       "  'id': 'CONVERTER_bed_to_fli_0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=CONVERTER_bed_to_fli_0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert BED to Feature Location Index',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_oddcomp.xml',\n",
       "  'description': 'Find protein sequence regions with a biased composition',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: oddcomp64/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+oddcomp64%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'oddcomp',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_pasteseq.xml',\n",
       "  'description': 'Insert one sequence into another',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: pasteseq66/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+pasteseq66%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'pasteseq',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/8bb5ea34809d/stacks/STACKS_sort_read_pairs.xml',\n",
       "  'description': 'Run the STACKS sort_read_pairs.pl and exec_velvet.pl wrappers',\n",
       "  'id': 'testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/STACKSassembleperead/1.2.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=testtoolshed.g2.bx.psu.edu%2Frepos%2Ftiagoantao%2Fstacks%2FSTACKSassembleperead%2F1.2.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'STACKS : assemble read pairs by locus',\n",
       "  'panel_section_id': 'ngs:_rads',\n",
       "  'panel_section_name': 'NGS: RADs',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '8bb5ea34809d',\n",
       "   'name': 'stacks',\n",
       "   'owner': 'tiagoantao',\n",
       "   'tool_shed': 'testtoolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.2.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/reldist.xml',\n",
       "  'description': 'calculate the distribution of relative distances',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_reldistbed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_reldistbed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'ReldistBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/7f3c448e119b/ncbi_blast_plus/tools/ncbi_blast_plus/ncbi_tblastx_wrapper.xml',\n",
       "  'description': 'Search translated nucleotide database with translated nucleotide query sequence(s)',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_tblastx_wrapper/0.1.07',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fncbi_blast_plus%2Fncbi_tblastx_wrapper%2F0.1.07',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'NCBI BLAST+ tblastx',\n",
       "  'panel_section_id': 'ngs:_mapping',\n",
       "  'panel_section_name': 'NGS: Mapping',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '7f3c448e119b',\n",
       "   'name': 'ncbi_blast_plus',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.1.07'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/maskFastaBed.xml',\n",
       "  'description': 'use intervals to mask sequences from a FASTA file',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_maskfastabed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_maskfastabed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MaskFastaBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/galaxy-central/tools/maf/maf_to_interval.xml',\n",
       "  'description': 'Converts a MAF formatted file to the Interval format',\n",
       "  'id': 'MAF_To_Interval1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=MAF_To_Interval1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MAF to Interval',\n",
       "  'panel_section_id': 'convert',\n",
       "  'panel_section_name': 'Convert Formats',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/fasta_to_bowtie_color_index_converter.xml',\n",
       "  'description': '',\n",
       "  'id': 'CONVERTER_fasta_to_bowtie_color_index',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=CONVERTER_fasta_to_bowtie_color_index',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert FASTA to Bowtie color space Index',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_sixpack.xml',\n",
       "  'description': 'Display a DNA sequence with 6-frame translation and ORFs',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: sixpack90/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+sixpack90%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'sixpack',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/genomespace/genomespace_file_browser_prod.xml',\n",
       "  'description': 'from file browser',\n",
       "  'id': 'genomespace_file_browser_prod',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=genomespace_file_browser_prod',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'GenomeSpace import',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '0.0.1'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/lped_to_pbed_converter.xml',\n",
       "  'description': '',\n",
       "  'id': 'lped2pbedconvert',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=lped2pbedconvert',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert lped to plink pbed',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '0.01'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_prettyseq.xml',\n",
       "  'description': 'Output sequence with translated ranges',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: prettyseq80/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+prettyseq80%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'prettyseq',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/maf/maf_filter.xml',\n",
       "  'description': 'by specified attributes',\n",
       "  'id': 'MAF_filter',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=MAF_filter',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Filter MAF',\n",
       "  'panel_section_id': 'fetchAlign',\n",
       "  'panel_section_name': 'Fetch Alignments',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_diffseq.xml',\n",
       "  'description': 'Find differences between nearly identical sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: diffseq22/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+diffseq22%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'diffseq',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/fasta_to_2bit.xml',\n",
       "  'description': '',\n",
       "  'id': 'CONVERTER_fasta_to_2bit',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=CONVERTER_fasta_to_2bit',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert FASTA to 2bit',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/7f3c448e119b/ncbi_blast_plus/tools/ncbi_blast_plus/ncbi_makeprofiledb.xml',\n",
       "  'description': 'Make profile database',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_makeprofiledb/0.1.07',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fncbi_blast_plus%2Fncbi_makeprofiledb%2F0.1.07',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'NCBI BLAST+ makeprofiledb',\n",
       "  'panel_section_id': 'ngs:_mapping',\n",
       "  'panel_section_name': 'NGS: Mapping',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '7f3c448e119b',\n",
       "   'name': 'ncbi_blast_plus',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.1.07'},\n",
       " {'config_file': '/shed_tools/testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/8bb5ea34809d/stacks/STACKS_sort_read_pairs_no_assembly.xml',\n",
       "  'description': 'Run STACKS sort_read_pairs.pl',\n",
       "  'id': 'testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/STACKSpairends/1.2.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=testtoolshed.g2.bx.psu.edu%2Frepos%2Ftiagoantao%2Fstacks%2FSTACKSpairends%2F1.2.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'STACKS : collate pair end sequences by locus',\n",
       "  'panel_section_id': 'ngs:_rads',\n",
       "  'panel_section_name': 'NGS: RADs',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '8bb5ea34809d',\n",
       "   'name': 'stacks',\n",
       "   'owner': 'tiagoantao',\n",
       "   'tool_shed': 'testtoolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.2.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/ucsc_gene_bed_to_exon_bed.xml',\n",
       "  'description': 'expander',\n",
       "  'id': 'gene2exon1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=gene2exon1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Gene BED To Exon/Intron/Codon BED',\n",
       "  'panel_section_id': 'features',\n",
       "  'panel_section_name': 'Extract Features',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/boris/filter_on_md/ac70bfaf1224/filter_on_md/MDtag_filter.xml',\n",
       "  'description': 'on MD tag string',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/boris/filter_on_md/MDtag_filter/1.0.2',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fboris%2Ffilter_on_md%2FMDtag_filter%2F1.0.2',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Filter mapped reads',\n",
       "  'panel_section_id': 'ngs:_samtools',\n",
       "  'panel_section_name': 'NGS: SAMtools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'ac70bfaf1224',\n",
       "   'name': 'filter_on_md',\n",
       "   'owner': 'boris',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.2'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_fuzzpro.xml',\n",
       "  'description': 'Protein pattern search',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: fuzzpro38/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+fuzzpro38%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'fuzzpro',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/7f3c448e119b/ncbi_blast_plus/tools/ncbi_blast_plus/blastxml_to_tabular.xml',\n",
       "  'description': 'Convert BLAST XML output to tabular',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/blastxml_to_tabular/0.1.07',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fncbi_blast_plus%2Fblastxml_to_tabular%2F0.1.07',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'BLAST XML to tabular',\n",
       "  'panel_section_id': 'ngs:_mapping',\n",
       "  'panel_section_name': 'NGS: Mapping',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '7f3c448e119b',\n",
       "   'name': 'ncbi_blast_plus',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.1.07'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/ucsc_tablebrowser_test.xml',\n",
       "  'description': 'table browser',\n",
       "  'id': 'ucsc_table_direct_test1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=ucsc_table_direct_test1',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'UCSC Test',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/288462ec2630/text_processing/awk.xml',\n",
       "  'description': 'with awk',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_awk_tool/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Ftext_processing%2Ftp_awk_tool%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Text reformatting',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '288462ec2630',\n",
       "   'name': 'text_processing',\n",
       "   'owner': 'bgruening',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/gatk2/84584664264c/gatk2/variant_validate.xml',\n",
       "  'description': '',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/gatk2/gatk2_variant_validate/2.8.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fgatk2%2Fgatk2_variant_validate%2F2.8.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Validate Variants',\n",
       "  'panel_section_id': 'ngs:_gatk',\n",
       "  'panel_section_name': 'NGS: GATK',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '84584664264c',\n",
       "   'name': 'gatk2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.8.0'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/wormbase.xml',\n",
       "  'description': 'server',\n",
       "  'id': 'wormbase',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=wormbase',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'WormBase',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_vectorstrip.xml',\n",
       "  'description': 'Strips out DNA between a pair of vector sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: vectorstrip106/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+vectorstrip106%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'vectorstrip',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_seqmatchall.xml',\n",
       "  'description': 'All-against-all comparison of a set of sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: seqmatchall83/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+seqmatchall83%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'seqmatchall',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_wordcount.xml',\n",
       "  'description': 'Counts words of a specified size in a DNA sequence',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: wordcount109/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+wordcount109%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'wordcount',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/makeWindowsBed.xml',\n",
       "  'description': 'make interval windows across a genome',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_makewindowsbed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_makewindowsbed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MakeWindowsBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/pasteWrapper.xml',\n",
       "  'description': 'two files side by side',\n",
       "  'id': 'Paste1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=Paste1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Paste',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/worm_modencode.xml',\n",
       "  'description': 'server',\n",
       "  'id': 'modENCODEworm',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=modENCODEworm',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'modENCODE worm',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/galaxy-central/tools/filters/grep.xml',\n",
       "  'description': 'lines that match an expression',\n",
       "  'id': 'Grep1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=Grep1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Select',\n",
       "  'panel_section_id': 'filter',\n",
       "  'panel_section_name': 'Filter and Sort',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/7f3c448e119b/ncbi_blast_plus/tools/ncbi_blast_plus/ncbi_blastdbcmd_wrapper.xml',\n",
       "  'description': 'Extract sequence(s) from BLAST database',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_blastdbcmd_wrapper/0.1.07',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fncbi_blast_plus%2Fncbi_blastdbcmd_wrapper%2F0.1.07',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'NCBI BLAST+ blastdbcmd entry(s)',\n",
       "  'panel_section_id': 'ngs:_mapping',\n",
       "  'panel_section_name': 'NGS: Mapping',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '7f3c448e119b',\n",
       "   'name': 'ncbi_blast_plus',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.1.07'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_SortSam.xml',\n",
       "  'description': 'sort SAM/BAM dataset',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_SortSam/1.136.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_SortSam%2F1.136.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'SortSam',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/expandBed.xml',\n",
       "  'description': 'replicate lines based on lists of values in columns',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_expandbed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_expandbed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'ExpandBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/koen/gbsx/3b325560887b/gbsx/gbsx_barcodeGenerator.xml',\n",
       "  'description': 'creates inline barcodes for a given enzyme',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/koen/gbsx/gbsx_barcodeGenerator/1.1.5',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fkoen%2Fgbsx%2Fgbsx_barcodeGenerator%2F1.1.5',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'GBSX: Barcode Generator',\n",
       "  'panel_section_id': 'ngs:_rads',\n",
       "  'panel_section_name': 'NGS: RADs',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '3b325560887b',\n",
       "   'name': 'gbsx',\n",
       "   'owner': 'koen',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.1.5'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_chips.xml',\n",
       "  'description': 'Codon usage statistics',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: chips10/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+chips10%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'chips',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_einverted.xml',\n",
       "  'description': 'Finds DNA inverted repeats',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: einverted28/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+einverted28%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'einverted',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/mergeBed.xml',\n",
       "  'description': 'combine overlapping/nearby intervals into a single interval',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_mergebed/2.24.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_mergebed%2F2.24.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MergeBED',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_nthseq.xml',\n",
       "  'description': 'Writes one sequence from a multiple set of sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: nthseq62/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+nthseq62%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'nthseq',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_checktrans.xml',\n",
       "  'description': 'Reports STOP codons and ORF statistics of a protein',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: checktrans9/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+checktrans9%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'checktrans',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/cuffcompare/b77178f66fc3/cuffcompare/cuffcompare_wrapper.xml',\n",
       "  'description': 'compare assembled transcripts to a reference annotation and track Cufflinks transcripts across multiple experiments',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/cuffcompare/cuffcompare/2.2.1.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fcuffcompare%2Fcuffcompare%2F2.2.1.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Cuffcompare',\n",
       "  'panel_section_id': 'ngs:_rna_analysis',\n",
       "  'panel_section_name': 'NGS: RNA Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b77178f66fc3',\n",
       "   'name': 'cuffcompare',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.2.1.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_cpgreport.xml',\n",
       "  'description': 'Reports all CpG rich regions',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: cpgreport16/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+cpgreport16%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'cpgreport',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/freebayes/99684adf84de/freebayes/freebayes.xml',\n",
       "  'description': '- bayesian genetic variant detector',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/freebayes/freebayes/0.4.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Ffreebayes%2Ffreebayes%2F0.4.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'FreeBayes',\n",
       "  'panel_section_id': 'ngs:_variant_analysis',\n",
       "  'panel_section_name': 'NGS: Variant Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '99684adf84de',\n",
       "   'name': 'freebayes',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.4.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_pepcoil.xml',\n",
       "  'description': 'Predicts coiled coil regions',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: pepcoil68/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+pepcoil68%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'pepcoil',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/tabular_to_fasta/0b4e36026794/tabular_to_fasta/tabular_to_fasta.xml',\n",
       "  'description': 'converts tabular file to FASTA format',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/tabular_to_fasta/tab2fasta/1.1.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Ftabular_to_fasta%2Ftab2fasta%2F1.1.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Tabular-to-FASTA',\n",
       "  'panel_section_id': 'fasta_manipulation',\n",
       "  'panel_section_name': 'FASTA manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '0b4e36026794',\n",
       "   'name': 'tabular_to_fasta',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.1.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_CollectBaseDistributionByCycle.xml',\n",
       "  'description': 'charts the nucleotide distribution per cycle in a SAM or BAM dataset',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_CollectBaseDistributionByCycle/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_CollectBaseDistributionByCycle%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'CollectBaseDistributionByCycle',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/bedToBam.xml',\n",
       "  'description': 'converter',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bedtobam/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_bedtobam%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'BED to BAM',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_NormalizeFasta.xml',\n",
       "  'description': 'normalize fasta datasets',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_NormalizeFasta/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_NormalizeFasta%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'NormalizeFasta',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/288462ec2630/text_processing/replace_text_in_column.xml',\n",
       "  'description': 'in a specific column',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_replace_in_column/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Ftext_processing%2Ftp_replace_in_column%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Replace Text',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '288462ec2630',\n",
       "   'name': 'text_processing',\n",
       "   'owner': 'bgruening',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/coverageBed.xml',\n",
       "  'description': 'of features in file A across the features in file B (coverageBed)',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_coveragebed/2.24.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_coveragebed%2F2.24.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Compute both the depth and breadth of coverage',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_charge.xml',\n",
       "  'description': 'Protein charge plot',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: charge8/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+charge8%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'charge',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/fisherBed.xml',\n",
       "  'description': 'calculate Fisher statistic between two feature files',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_fisher/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_fisher%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'FisherBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/b723c120161a/sra_tools/sra_pileup.xml',\n",
       "  'description': 'from NCBI sra.',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/sra_pileup/1.2.5',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsra_tools%2Fsra_pileup%2F1.2.5',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Generate pileup format',\n",
       "  'panel_section_id': 'ncbi_sra_tools',\n",
       "  'panel_section_name': 'NCBI SRA Tools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b723c120161a',\n",
       "   'name': 'sra_tools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.2.5'},\n",
       " {'config_file': '/galaxy-central/tools/filters/axt_to_fasta.xml',\n",
       "  'description': 'Converts an AXT formatted file to FASTA format',\n",
       "  'id': 'axt_to_fasta',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=axt_to_fasta',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'AXT to FASTA',\n",
       "  'panel_section_id': 'convert',\n",
       "  'panel_section_name': 'Convert Formats',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_coderet.xml',\n",
       "  'description': 'Extract CDS, mRNA and translations from feature tables',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: coderet13/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+coderet13%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'coderet',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/gatk2/84584664264c/gatk2/reduce_reads.xml',\n",
       "  'description': 'in BAM files',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/gatk2/gatk2_reduce_reads/2.8.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fgatk2%2Fgatk2_reduce_reads%2F2.8.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Reduce Reads',\n",
       "  'panel_section_id': 'ngs:_gatk',\n",
       "  'panel_section_name': 'NGS: GATK',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '84584664264c',\n",
       "   'name': 'gatk2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.8.0'},\n",
       " {'config_file': '/galaxy-central/tools/maf/maf_to_fasta.xml',\n",
       "  'description': 'Converts a MAF formatted file to FASTA format',\n",
       "  'id': 'MAF_To_Fasta1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=MAF_To_Fasta1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MAF to FASTA',\n",
       "  'panel_section_id': 'convert',\n",
       "  'panel_section_name': 'Convert Formats',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/ratmine.xml',\n",
       "  'description': 'server',\n",
       "  'id': 'ratmine',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=ratmine',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'Ratmine',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_degapseq.xml',\n",
       "  'description': 'Removes gap characters from sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: degapseq20/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+degapseq20%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'degapseq',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/ucsc_tablebrowser_archaea.xml',\n",
       "  'description': 'table browser',\n",
       "  'id': 'ucsc_table_direct_archaea1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=ucsc_table_direct_archaea1',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'UCSC Archaea',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_marscan.xml',\n",
       "  'description': 'Finds MAR/SAR sites in nucleic sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: marscan49/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+marscan49%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'marscan',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/gatk2/84584664264c/gatk2/variant_eval.xml',\n",
       "  'description': '',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/gatk2/gatk2_variant_eval/2.8.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fgatk2%2Fgatk2_variant_eval%2F2.8.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Eval Variants',\n",
       "  'panel_section_id': 'ngs:_gatk',\n",
       "  'panel_section_name': 'NGS: GATK',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '84584664264c',\n",
       "   'name': 'gatk2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.8.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/288462ec2630/text_processing/grep.xml',\n",
       "  'description': '(grep)',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_grep_tool/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Ftext_processing%2Ftp_grep_tool%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Search in textfiles',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '288462ec2630',\n",
       "   'name': 'text_processing',\n",
       "   'owner': 'bgruening',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_maskfeat.xml',\n",
       "  'description': 'Mask off features of a sequence',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: maskfeat50/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+maskfeat50%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'maskfeat',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/upload.xml',\n",
       "  'description': 'from your computer',\n",
       "  'id': 'upload1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=upload1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Upload File',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.1.4'},\n",
       " {'config_file': '/galaxy-central/tools/filters/headWrapper.xml',\n",
       "  'description': 'lines from a dataset',\n",
       "  'id': 'Show beginning1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=Show+beginning1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Select first',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_supermatcher.xml',\n",
       "  'description': 'Match large sequences against one or more other sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: supermatcher95/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+supermatcher95%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'supermatcher',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/48b5f719e36a/str_fm/microsatellite.xml',\n",
       "  'description': 'for short read, reference, and mapped data',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/microsatellite/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Farkarachai-fungtammasan%2Fstr_fm%2Fmicrosatellite%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'STR detection',\n",
       "  'panel_section_id': 'str-fm:_microsatellite_analysis',\n",
       "  'panel_section_name': 'STR-FM: Microsatellite Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '48b5f719e36a',\n",
       "   'name': 'str_fm',\n",
       "   'owner': 'arkarachai-fungtammasan',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/fly_modencode.xml',\n",
       "  'description': 'server',\n",
       "  'id': 'modENCODEfly',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=modENCODEfly',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'modENCODE fly',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/velvet/08256557922f/velvet/velvetg.xml',\n",
       "  'description': 'Velvet sequence assembler for very short reads',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/velvet/velvetg/1.2.10.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fvelvet%2Fvelvetg%2F1.2.10.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'velvetg',\n",
       "  'panel_section_id': 'ngs:_de_novo',\n",
       "  'panel_section_name': 'NGS: De Novo',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '08256557922f',\n",
       "   'name': 'velvet',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.2.10.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_equicktandem.xml',\n",
       "  'description': 'Finds tandem repeats',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: equicktandem31/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+equicktandem31%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'equicktandem',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/macs2/6d4babad010f/macs2/macs2_filterdup.xml',\n",
       "  'description': 'Remove duplicate reads at the same position',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_filterdup/2.1.0.20151222.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fmacs2%2Fmacs2_filterdup%2F2.1.0.20151222.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MACS2 filterdup',\n",
       "  'panel_section_id': 'ngs:_peak_calling',\n",
       "  'panel_section_name': 'NGS: Peak Calling',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '6d4babad010f',\n",
       "   'name': 'macs2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.1.0.20151222.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_tranalign.xml',\n",
       "  'description': 'Align nucleic coding regions given the aligned proteins',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: tranalign100/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+tranalign100%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'tranalign',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_cai_custom.xml',\n",
       "  'description': 'CAI codon adaptation index using custom codon usage file',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: cai_custom6/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+cai_custom6%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'cai custom',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/288462ec2630/text_processing/tail.xml',\n",
       "  'description': 'lines from a dataset (tail)',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_tail_tool/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Ftext_processing%2Ftp_tail_tool%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Select last',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '288462ec2630',\n",
       "   'name': 'text_processing',\n",
       "   'owner': 'bgruening',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_noreturn.xml',\n",
       "  'description': 'Removes carriage return from ASCII files',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: noreturn60/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+noreturn60%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'noreturn',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/48b5f719e36a/str_fm/GenotypingSTR.xml',\n",
       "  'description': 'that occur during sequencing and library prep',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/GenotypeSTR/2.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Farkarachai-fungtammasan%2Fstr_fm%2FGenotypeSTR%2F2.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Correct genotype for STR errors',\n",
       "  'panel_section_id': 'str-fm:_microsatellite_analysis',\n",
       "  'panel_section_name': 'STR-FM: Microsatellite Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '48b5f719e36a',\n",
       "   'name': 'str_fm',\n",
       "   'owner': 'arkarachai-fungtammasan',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/7f3c448e119b/ncbi_blast_plus/tools/ncbi_blast_plus/ncbi_convert2blastmask_wrapper.xml',\n",
       "  'description': 'Convert masking information in lower-case masked FASTA input to file formats suitable for makeblastdb',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_convert2blastmask_wrapper/0.1.07',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fncbi_blast_plus%2Fncbi_convert2blastmask_wrapper%2F0.1.07',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'NCBI BLAST+ convert2blastmask',\n",
       "  'panel_section_id': 'ngs:_mapping',\n",
       "  'panel_section_name': 'NGS: Mapping',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '7f3c448e119b',\n",
       "   'name': 'ncbi_blast_plus',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.1.07'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/288462ec2630/text_processing/cat.xml',\n",
       "  'description': 'tail-to-head (cat)',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cat/0.1.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Ftext_processing%2Ftp_cat%2F0.1.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Concatenate datasets',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '288462ec2630',\n",
       "   'name': 'text_processing',\n",
       "   'owner': 'bgruening',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.1.0'},\n",
       " {'config_file': '/galaxy-central/tools/maf/genebed_maf_to_fasta.xml',\n",
       "  'description': 'given a set of coding exon intervals',\n",
       "  'id': 'GeneBed_Maf_Fasta2',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=GeneBed_Maf_Fasta2',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Stitch Gene blocks',\n",
       "  'panel_section_id': 'fetchAlign',\n",
       "  'panel_section_name': 'Fetch Alignments',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/overlapBed.xml',\n",
       "  'description': 'computes the amount of overlap from two intervals',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_overlapbed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_overlapbed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'OverlapBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/trimmer.xml',\n",
       "  'description': 'leading or trailing characters',\n",
       "  'id': 'trimmer',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=trimmer',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Trim',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '0.0.1'},\n",
       " {'config_file': '/galaxy-central/tools/filters/CreateInterval.xml',\n",
       "  'description': 'as a new dataset',\n",
       "  'id': 'createInterval',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=createInterval',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Create single interval',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/fastq_quality_filter/e41385662e5e/fastq_quality_filter/fastq_quality_filter.xml',\n",
       "  'description': '',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/fastq_quality_filter/cshl_fastq_quality_filter/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Ffastq_quality_filter%2Fcshl_fastq_quality_filter%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Filter by quality',\n",
       "  'panel_section_id': 'ngs:_qc_and_manipulation',\n",
       "  'panel_section_name': 'NGS: QC and manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'e41385662e5e',\n",
       "   'name': 'fastq_quality_filter',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_preg.xml',\n",
       "  'description': 'Regular expression search of a protein sequence',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: preg78/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+preg78%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'preg',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/data_manager_gatk_picard_index_builder/700f2df51eb0/data_manager_gatk_picard_index_builder/data_manager/data_manager_gatk_picard_index_builder.xml',\n",
       "  'description': 'builder',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/data_manager_gatk_picard_index_builder/gatk_picard_index_builder/0.0.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fdata_manager_gatk_picard_index_builder%2Fgatk_picard_index_builder%2F0.0.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'DataManagerTool',\n",
       "  'name': 'Generate GATK-sorted Picard indexes',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '700f2df51eb0',\n",
       "   'name': 'data_manager_gatk_picard_index_builder',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/gatk2/84584664264c/gatk2/variant_recalibrator.xml',\n",
       "  'description': '',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/gatk2/gatk2_variant_recalibrator/2.8.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fgatk2%2Fgatk2_variant_recalibrator%2F2.8.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Variant Recalibrator',\n",
       "  'panel_section_id': 'ngs:_gatk',\n",
       "  'panel_section_name': 'NGS: GATK',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '84584664264c',\n",
       "   'name': 'gatk2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.8.0'},\n",
       " {'config_file': '/galaxy-central/tools/stats/grouping.xml',\n",
       "  'description': 'data by a column and perform aggregate operation on other columns.',\n",
       "  'id': 'Grouping1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=Grouping1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Group',\n",
       "  'panel_section_id': 'group',\n",
       "  'panel_section_name': 'Join, Subtract and Group',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '2.1.1'},\n",
       " {'config_file': '/shed_tools/testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/8bb5ea34809d/stacks/STACKS_rxstacks.xml',\n",
       "  'description': 'Run the rxstacks tool',\n",
       "  'id': 'testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/STACKSrxstacks/1.2.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=testtoolshed.g2.bx.psu.edu%2Frepos%2Ftiagoantao%2Fstacks%2FSTACKSrxstacks%2F1.2.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'STACKS : rxstacks',\n",
       "  'panel_section_id': 'ngs:_rads',\n",
       "  'panel_section_name': 'NGS: RADs',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '8bb5ea34809d',\n",
       "   'name': 'stacks',\n",
       "   'owner': 'tiagoantao',\n",
       "   'tool_shed': 'testtoolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.2.0'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/microbial_import.xml',\n",
       "  'description': '',\n",
       "  'id': 'microbial_import1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=microbial_import1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Get Microbial Data',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/clusterBed.xml',\n",
       "  'description': 'cluster overlapping/nearby intervals',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_clusterbed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_clusterbed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'ClusterBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/macs2/6d4babad010f/macs2/macs2_bdgdiff.xml',\n",
       "  'description': 'Differential peak detection based on paired four bedgraph files',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_bdgdiff/2.1.0.20151222.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fmacs2%2Fmacs2_bdgdiff%2F2.1.0.20151222.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MACS2 bdgdiff',\n",
       "  'panel_section_id': 'ngs:_peak_calling',\n",
       "  'panel_section_name': 'NGS: Peak Calling',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '6d4babad010f',\n",
       "   'name': 'macs2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.1.0.20151222.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_FilterSamReads.xml',\n",
       "  'description': 'include or exclude aligned and unaligned reads and read lists',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_FilterSamReads/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_FilterSamReads%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'FilterSamReads',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/flymine.xml',\n",
       "  'description': 'server',\n",
       "  'id': 'flymine',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=flymine',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'Flymine',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_transeq.xml',\n",
       "  'description': 'Translate nucleic acid sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: transeq101/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+transeq101%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'transeq',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/macs2/6d4babad010f/macs2/macs2_bdgbroadcall.xml',\n",
       "  'description': 'Call broad peaks from bedGraph output',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_bdgbroadcall/2.1.0.20151222.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fmacs2%2Fmacs2_bdgbroadcall%2F2.1.0.20151222.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MACS2 bdgbroadcall',\n",
       "  'panel_section_id': 'ngs:_peak_calling',\n",
       "  'panel_section_name': 'NGS: Peak Calling',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '6d4babad010f',\n",
       "   'name': 'macs2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.1.0.20151222.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/linear_regression/cf431604ec3e/linear_regression/linear_regression.xml',\n",
       "  'description': '',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/linear_regression/LinearRegression1/1.0.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Flinear_regression%2FLinearRegression1%2F1.0.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Perform Linear Regression',\n",
       "  'panel_section_id': 'multiple_regression',\n",
       "  'panel_section_name': 'Multiple Regression',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'cf431604ec3e',\n",
       "   'name': 'linear_regression',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_EstimateLibraryComplexity.xml',\n",
       "  'description': 'assess sequence library complexity from read sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_EstimateLibraryComplexity/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_EstimateLibraryComplexity%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'EstimateLibraryComplexity',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/velvet/08256557922f/velvet/velveth.xml',\n",
       "  'description': 'Prepare a dataset for the Velvet velvetg Assembler',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/velvet/velveth/1.2.10.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fvelvet%2Fvelveth%2F1.2.10.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'velveth',\n",
       "  'panel_section_id': 'ngs:_de_novo',\n",
       "  'panel_section_name': 'NGS: De Novo',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '08256557922f',\n",
       "   'name': 'velvet',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.2.10.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/48b5f719e36a/str_fm/PEsortedSAM2readprofile.xml',\n",
       "  'description': 'and get the reference STR allele from the reference genome',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/PEsortedSAM2readprofile/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Farkarachai-fungtammasan%2Fstr_fm%2FPEsortedSAM2readprofile%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Combine mapped faux paired-end reads',\n",
       "  'panel_section_id': 'str-fm:_microsatellite_analysis',\n",
       "  'panel_section_name': 'STR-FM: Microsatellite Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '48b5f719e36a',\n",
       "   'name': 'str_fm',\n",
       "   'owner': 'arkarachai-fungtammasan',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/bed_to_gff_converter.xml',\n",
       "  'description': '',\n",
       "  'id': 'CONVERTER_bed_to_gff_0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=CONVERTER_bed_to_gff_0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert BED to GFF',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '2.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/hisat2/c37b4554f794/hisat2/hisat2.xml',\n",
       "  'description': 'A fast and sensitive alignment program',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/hisat2/hisat2/1.0.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fhisat2%2Fhisat2%2F1.0.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'HISAT2',\n",
       "  'panel_section_id': 'ngs:_rna_analysis',\n",
       "  'panel_section_name': 'NGS: RNA Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b1e25f9b5eab',\n",
       "   'name': 'hisat2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_seqret.xml',\n",
       "  'description': 'Reads and writes sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: seqret84/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+seqret84%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'seqret',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_getorf.xml',\n",
       "  'description': 'Finds and extracts open reading frames (ORFs)',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: getorf42/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+getorf42%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'getorf',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/gatk2/84584664264c/gatk2/indel_realigner.xml',\n",
       "  'description': '- perform local realignment',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/gatk2/gatk2_indel_realigner/2.8.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fgatk2%2Fgatk2_indel_realigner%2F2.8.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Indel Realigner',\n",
       "  'panel_section_id': 'ngs:_gatk',\n",
       "  'panel_section_name': 'NGS: GATK',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '84584664264c',\n",
       "   'name': 'gatk2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.8.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/288462ec2630/text_processing/sorted_uniq.xml',\n",
       "  'description': 'assuming sorted input file',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_uniq_tool/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Ftext_processing%2Ftp_uniq_tool%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Unique lines',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '288462ec2630',\n",
       "   'name': 'text_processing',\n",
       "   'owner': 'bgruening',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/gatk2/84584664264c/gatk2/realigner_target_creator.xml',\n",
       "  'description': 'for use in local realignment',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/gatk2/gatk2_realigner_target_creator/2.8.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fgatk2%2Fgatk2_realigner_target_creator%2F2.8.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Realigner Target Creator',\n",
       "  'panel_section_id': 'ngs:_gatk',\n",
       "  'panel_section_name': 'NGS: GATK',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '84584664264c',\n",
       "   'name': 'gatk2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.8.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_showfeat.xml',\n",
       "  'description': 'Show features of a sequence',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: showfeat85/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+showfeat85%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'showfeat',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/secure_hash_message_digest.xml',\n",
       "  'description': 'on a dataset',\n",
       "  'id': 'secure_hash_message_digest',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=secure_hash_message_digest',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Secure Hash / Message Digest',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '0.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_QualityScoreDistribution.xml',\n",
       "  'description': 'chart quality score distribution',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_QualityScoreDistribution/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_QualityScoreDistribution%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'QualityScoreDistribution',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/gff2bed.xml',\n",
       "  'description': 'converter',\n",
       "  'id': 'gff2bed1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=gff2bed1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'GFF-to-BED',\n",
       "  'panel_section_id': 'convert',\n",
       "  'panel_section_name': 'Convert Formats',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/maf_to_fasta_converter.xml',\n",
       "  'description': '',\n",
       "  'id': 'CONVERTER_maf_to_fasta_0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=CONVERTER_maf_to_fasta_0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert MAF to Fasta',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/galaxy-central/tools/maf/maf_limit_size.xml',\n",
       "  'description': 'by Size',\n",
       "  'id': 'maf_limit_size1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=maf_limit_size1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Filter MAF blocks',\n",
       "  'panel_section_id': 'fetchAlign',\n",
       "  'panel_section_name': 'Fetch Alignments',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_cai.xml',\n",
       "  'description': 'CAI codon adaptation index',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: cai6/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+cai6%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'cai',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/convert_characters.xml',\n",
       "  'description': 'delimiters to TAB',\n",
       "  'id': 'Convert characters1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=Convert+characters1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_dreg.xml',\n",
       "  'description': 'Regular expression search of a nucleotide sequence',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: dreg27/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+dreg27%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'dreg',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_twofeat.xml',\n",
       "  'description': 'Finds neighbouring pairs of features in sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: twofeat104/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+twofeat104%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'twofeat',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_btwisted.xml',\n",
       "  'description': 'Calculates the twisting in a B-DNA sequence',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: btwisted5/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+btwisted5%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'btwisted',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/sam_to_bam/881e16ad05c6/sam_to_bam/sam_to_bam.xml',\n",
       "  'description': 'convert SAM to BAM',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/sam_to_bam/sam_to_bam/2.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsam_to_bam%2Fsam_to_bam%2F2.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'SAM-to-BAM',\n",
       "  'panel_section_id': 'ngs:_qc_and_manipulation',\n",
       "  'panel_section_name': 'NGS: QC and manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '881e16ad05c6',\n",
       "   'name': 'sam_to_bam',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/288462ec2630/text_processing/unsorted_uniq.xml',\n",
       "  'description': 'occurrences of each record',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sorted_uniq/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Ftext_processing%2Ftp_sorted_uniq%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Unique',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '288462ec2630',\n",
       "   'name': 'text_processing',\n",
       "   'owner': 'bgruening',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_newseq.xml',\n",
       "  'description': 'Type in a short new sequence',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: newseq59/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+newseq59%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'newseq',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/getfastaBed.xml',\n",
       "  'description': 'use intervals to extract sequences from a FASTA file',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_getfastabed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_getfastabed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'GetFastaBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_plotcon.xml',\n",
       "  'description': 'Plot quality of conservation of a sequence alignment',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: plotcon75/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+plotcon75%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'plotcon',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/fasta_to_tabular_converter.xml',\n",
       "  'description': '',\n",
       "  'id': 'CONVERTER_fasta_to_tabular',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=CONVERTER_fasta_to_tabular',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert FASTA to Tabular',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/48b5f719e36a/str_fm/space2underscore_readname.xml',\n",
       "  'description': '--change space to underscore in the read name column',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/space2underscore_readname/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Farkarachai-fungtammasan%2Fstr_fm%2Fspace2underscore_readname%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Read name modifier',\n",
       "  'panel_section_id': 'str-fm:_microsatellite_analysis',\n",
       "  'panel_section_name': 'STR-FM: Microsatellite Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '48b5f719e36a',\n",
       "   'name': 'str_fm',\n",
       "   'owner': 'arkarachai-fungtammasan',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/7f3c448e119b/ncbi_blast_plus/tools/ncbi_blast_plus/ncbi_rpsblast_wrapper.xml',\n",
       "  'description': 'Search protein domain database (PSSMs) with protein query sequence(s)',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_rpsblast_wrapper/0.1.07',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fncbi_blast_plus%2Fncbi_rpsblast_wrapper%2F0.1.07',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'NCBI BLAST+ rpsblast',\n",
       "  'panel_section_id': 'ngs:_mapping',\n",
       "  'panel_section_name': 'NGS: Mapping',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '7f3c448e119b',\n",
       "   'name': 'ncbi_blast_plus',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.1.07'},\n",
       " {'config_file': '/galaxy-central/tools/maf/maf_limit_to_species.xml',\n",
       "  'description': 'by Species',\n",
       "  'id': 'MAF_Limit_To_Species1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=MAF_Limit_To_Species1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Filter MAF blocks',\n",
       "  'panel_section_id': 'fetchAlign',\n",
       "  'panel_section_name': 'Fetch Alignments',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/bamToFastq.xml',\n",
       "  'description': '',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtofastq/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_bamtofastq%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert from BAM to FastQ',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/randomBed.xml',\n",
       "  'description': 'generate random intervals in a genome',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_randombed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_randombed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'RandomBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/macs2/6d4babad010f/macs2/macs2_randsample.xml',\n",
       "  'description': 'Randomly sample number or percentage of total reads',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_randsample/2.1.0.20151222.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fmacs2%2Fmacs2_randsample%2F2.1.0.20151222.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MACS2 randsample',\n",
       "  'panel_section_id': 'ngs:_peak_calling',\n",
       "  'panel_section_name': 'NGS: Peak Calling',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '6d4babad010f',\n",
       "   'name': 'macs2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.1.0.20151222.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/rnateam/mafft/f0606dfd5195/mafft/mafft.xml',\n",
       "  'description': 'Multiple alignment program for amino acid or nucleotide sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/rnateam/mafft/rbc_mafft/7.221.3',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Frnateam%2Fmafft%2Frbc_mafft%2F7.221.3',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MAFFT',\n",
       "  'panel_section_id': 'multiple_alignments',\n",
       "  'panel_section_name': 'Multiple Alignments',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f0606dfd5195',\n",
       "   'name': 'mafft',\n",
       "   'owner': 'rnateam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '7.221.3'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/288462ec2630/text_processing/recurring_lines.xml',\n",
       "  'description': 'with recurring lines',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_text_file_with_recurring_lines/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Ftext_processing%2Ftp_text_file_with_recurring_lines%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Create text file',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '288462ec2630',\n",
       "   'name': 'text_processing',\n",
       "   'owner': 'bgruening',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/zebrafishmine.xml',\n",
       "  'description': 'server',\n",
       "  'id': 'zebrafishmine',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=zebrafishmine',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'ZebrafishMine',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_garnier.xml',\n",
       "  'description': 'Predicts protein secondary structure',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: garnier40/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+garnier40%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'garnier',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_SamToFastq.xml',\n",
       "  'description': 'extract reads and qualities from SAM/BAM dataset and convert to fastq',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_SamToFastq/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_SamToFastq%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'SamToFastq',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_trimest.xml',\n",
       "  'description': 'Trim poly-A tails off EST sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: trimest102/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+trimest102%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'trimest',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_dotpath.xml',\n",
       "  'description': 'Non-overlapping wordmatch dotplot of two sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: dotpath25/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+dotpath25%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'dotpath',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/linksBed.xml',\n",
       "  'description': 'create a HTML page of links to UCSC locations',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_links/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_links%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'LinksBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_FastqToSam.xml',\n",
       "  'description': 'convert Fastq data into unaligned BAM',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_FastqToSam/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_FastqToSam%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'FastqToSam',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/galaxy-central/tools/maf/maf_split_by_species.xml',\n",
       "  'description': 'by Species',\n",
       "  'id': 'MAF_split_blocks_by_species1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=MAF_split_blocks_by_species1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Split MAF blocks',\n",
       "  'panel_section_id': 'fetchAlign',\n",
       "  'panel_section_name': 'Fetch Alignments',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/peterjc/mummer/5f5a358b281f/mummer/tools/mummer/mummer.xml',\n",
       "  'description': 'Combine mummer/nucmer/promer with mummerplot',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/peterjc/mummer/mummerplot_wrapper/0.0.5',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fpeterjc%2Fmummer%2Fmummerplot_wrapper%2F0.0.5',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MUMmer dotplot',\n",
       "  'panel_section_id': 'multiple_alignments',\n",
       "  'panel_section_name': 'Multiple Alignments',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '5f5a358b281f',\n",
       "   'name': 'mummer',\n",
       "   'owner': 'peterjc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.0.5'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/xuebing/meme/547a939f9217/meme/meme.xml',\n",
       "  'description': 'motif discovery',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/xuebing/meme/meme_meme/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fxuebing%2Fmeme%2Fmeme_meme%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MEME',\n",
       "  'panel_section_id': 'motif_tools',\n",
       "  'panel_section_name': 'Motif Tools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '547a939f9217',\n",
       "   'name': 'meme',\n",
       "   'owner': 'xuebing',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_CollectGcBiasMetrics.xml',\n",
       "  'description': 'charts the GC bias metrics',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_CollectGcBiasMetrics/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_CollectGcBiasMetrics%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'CollectGcBiasMetrics',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/lav_to_bed.xml',\n",
       "  'description': 'Converts a LAV formatted file to BED format',\n",
       "  'id': 'lav_to_bed1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=lav_to_bed1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'LAV to BED',\n",
       "  'panel_section_id': 'convert',\n",
       "  'panel_section_name': 'Convert Formats',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/compare.xml',\n",
       "  'description': 'to find common or distinct rows',\n",
       "  'id': 'comp1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=comp1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Compare two Datasets',\n",
       "  'panel_section_id': 'group',\n",
       "  'panel_section_name': 'Join, Subtract and Group',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.2'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/data_manager_bwa_index_builder/cc5ae94acf01/data_manager_bwa_index_builder/data_manager/bwa_color_space_index_builder.xml',\n",
       "  'description': 'builder',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/data_manager_bwa_index_builder/bwa_color_space_index_builder_data_manager/0.0.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fdata_manager_bwa_index_builder%2Fbwa_color_space_index_builder_data_manager%2F0.0.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'DataManagerTool',\n",
       "  'name': 'BWA Color index',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'cc5ae94acf01',\n",
       "   'name': 'data_manager_bwa_index_builder',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_AddCommentsToBam.xml',\n",
       "  'description': 'add comments to BAM dataset',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_AddCommentsToBam/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_AddCommentsToBam%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'AddCommentsToBam',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/shed_tools/testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/8bb5ea34809d/stacks/STACKS_procrad.xml',\n",
       "  'description': 'Run the STACKS cleaning script',\n",
       "  'id': 'testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/STACKSprocrad/1.2.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=testtoolshed.g2.bx.psu.edu%2Frepos%2Ftiagoantao%2Fstacks%2FSTACKSprocrad%2F1.2.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'STACKS : Process radtags',\n",
       "  'panel_section_id': 'ngs:_rads',\n",
       "  'panel_section_name': 'NGS: RADs',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '8bb5ea34809d',\n",
       "   'name': 'stacks',\n",
       "   'owner': 'tiagoantao',\n",
       "   'tool_shed': 'testtoolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.2.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_polydot.xml',\n",
       "  'description': 'Displays all-against-all dotplots of a set of sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: polydot77/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+polydot77%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'polydot',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/windowBed.xml',\n",
       "  'description': 'find overlapping intervals within a window around an interval',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_windowbed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_windowbed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'WindowBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_dan.xml',\n",
       "  'description': 'Calculates DNA RNA/DNA melting temperature',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: dan19/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+dan19%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'dan',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_sirna.xml',\n",
       "  'description': 'Finds siRNA duplexes in mRNA',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: sirna89/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+sirna89%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'sirna',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/sam_merge/1977f1637890/sam_merge/sam_merge.xml',\n",
       "  'description': 'merges BAM files together',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/sam_merge/sam_merge2/1.2.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fsam_merge%2Fsam_merge2%2F1.2.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Merge BAM Files',\n",
       "  'panel_section_id': 'ngs:_samtools',\n",
       "  'panel_section_name': 'NGS: SAMtools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '1977f1637890',\n",
       "   'name': 'sam_merge',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.2.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_est2genome.xml',\n",
       "  'description': 'Align EST and genomic DNA sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: est2genome32/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+est2genome32%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'est2genome',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/8bb5ea34809d/stacks/STACKS_genotypes.xml',\n",
       "  'description': 'Run the STACKS genotypes program',\n",
       "  'id': 'testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/STACKSgenotypes/1.2.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=testtoolshed.g2.bx.psu.edu%2Frepos%2Ftiagoantao%2Fstacks%2FSTACKSgenotypes%2F1.2.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'STACKS : genotypes',\n",
       "  'panel_section_id': 'ngs:_rads',\n",
       "  'panel_section_name': 'NGS: RADs',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '8bb5ea34809d',\n",
       "   'name': 'stacks',\n",
       "   'owner': 'tiagoantao',\n",
       "   'tool_shed': 'testtoolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.2.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/fixedValueColumn.xml',\n",
       "  'description': 'to an existing dataset',\n",
       "  'id': 'addValue',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=addValue',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Add column',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/bamtools_split/4555f451dd79/bamtools_split/bamtools-split.xml',\n",
       "  'description': 'BAM datasets on variety of attributes',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/bamtools_split/bamSplit/0.0.2',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fbamtools_split%2FbamSplit%2F0.0.2',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Split',\n",
       "  'panel_section_id': 'ngs:_bamtools',\n",
       "  'panel_section_name': 'NGS: BamTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '4555f451dd79',\n",
       "   'name': 'bamtools_split',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.0.2'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/shuffleBed.xml',\n",
       "  'description': 'randomly redistrubute intervals in a genome',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_shufflebed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_shufflebed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'ShuffleBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/freebayes/99684adf84de/freebayes/leftalign.xml',\n",
       "  'description': 'indels in BAM datasets',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/freebayes/bamleftalign/0.4',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Ffreebayes%2Fbamleftalign%2F0.4',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'BamLeftAlign',\n",
       "  'panel_section_id': 'ngs:_variant_analysis',\n",
       "  'panel_section_name': 'NGS: Variant Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '99684adf84de',\n",
       "   'name': 'freebayes',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.4'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_freak.xml',\n",
       "  'description': 'Residue/base frequency table or plot',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: freak36/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+freak36%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'freak',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/flymine_test.xml',\n",
       "  'description': 'server',\n",
       "  'id': 'flymine_test',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=flymine_test',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'Flymine test',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/7f3c448e119b/ncbi_blast_plus/tools/ncbi_blast_plus/ncbi_blastx_wrapper.xml',\n",
       "  'description': 'Search protein database with translated nucleotide query sequence(s)',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_blastx_wrapper/0.1.07',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fncbi_blast_plus%2Fncbi_blastx_wrapper%2F0.1.07',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'NCBI BLAST+ blastx',\n",
       "  'panel_section_id': 'ngs:_mapping',\n",
       "  'panel_section_name': 'NGS: Mapping',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '7f3c448e119b',\n",
       "   'name': 'ncbi_blast_plus',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.1.07'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_pepwindow.xml',\n",
       "  'description': 'Displays protein hydropathy',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: pepwindow73/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+pepwindow73%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'pepwindow',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/7f3c448e119b/ncbi_blast_plus/tools/ncbi_blast_plus/ncbi_blastp_wrapper.xml',\n",
       "  'description': 'Search protein database with protein query sequence(s)',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_blastp_wrapper/0.1.07',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fncbi_blast_plus%2Fncbi_blastp_wrapper%2F0.1.07',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'NCBI BLAST+ blastp',\n",
       "  'panel_section_id': 'ngs:_mapping',\n",
       "  'panel_section_name': 'NGS: Mapping',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '7f3c448e119b',\n",
       "   'name': 'ncbi_blast_plus',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.1.07'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/data_manager_bowtie2_index_builder/e87aeff2cf88/data_manager_bowtie2_index_builder/data_manager/bowtie2_index_builder.xml',\n",
       "  'description': 'builder',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/data_manager_bowtie2_index_builder/bowtie2_index_builder_data_manager/2.2.6',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fdata_manager_bowtie2_index_builder%2Fbowtie2_index_builder_data_manager%2F2.2.6',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'DataManagerTool',\n",
       "  'name': 'Bowtie2 index',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'e87aeff2cf88',\n",
       "   'name': 'data_manager_bowtie2_index_builder',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.2.6'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/mousemine.xml',\n",
       "  'description': 'server',\n",
       "  'id': 'mousemine',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=mousemine',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'MouseMine',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/joiner.xml',\n",
       "  'description': 'side by side on a specified field',\n",
       "  'id': 'join1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=join1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Join two Datasets',\n",
       "  'panel_section_id': 'group',\n",
       "  'panel_section_name': 'Join, Subtract and Group',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '2.0.2'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_pepwindowall.xml',\n",
       "  'description': 'Displays protein hydropathy of a set of sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: pepwindowall74/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+pepwindowall74%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'pepwindowall',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/cbi_rice_mart.xml',\n",
       "  'description': 'rice mart',\n",
       "  'id': 'cbi_rice_mart',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=cbi_rice_mart',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'CBI Rice Mart',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/fasta_to_len.xml',\n",
       "  'description': '',\n",
       "  'id': 'CONVERTER_fasta_to_len',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=CONVERTER_fasta_to_len',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert FASTA to len file',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/288462ec2630/text_processing/cut.xml',\n",
       "  'description': 'columns from a table (cut)',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_cut_tool/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Ftext_processing%2Ftp_cut_tool%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Cut',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '288462ec2630',\n",
       "   'name': 'text_processing',\n",
       "   'owner': 'bgruening',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/ebi_sra.xml',\n",
       "  'description': 'ENA SRA',\n",
       "  'id': 'ebi_sra_main',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=ebi_sra_main',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'EBI SRA',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_fuzztran.xml',\n",
       "  'description': 'Protein pattern search after translation',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: fuzztran39/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+fuzztran39%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'fuzztran',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_MergeBamAlignment.xml',\n",
       "  'description': 'merge alignment data with additional info stored in an unmapped BAM dataset',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MergeBamAlignment/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_MergeBamAlignment%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MergeBamAlignment',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/galaxy-central/tools/maf/interval_maf_to_merged_fasta.xml',\n",
       "  'description': 'given a set of genomic intervals',\n",
       "  'id': 'Interval_Maf_Merged_Fasta2',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=Interval_Maf_Merged_Fasta2',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Stitch MAF blocks',\n",
       "  'panel_section_id': 'fetchAlign',\n",
       "  'panel_section_name': 'Fetch Alignments',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_newcpgseek.xml',\n",
       "  'description': 'Reports CpG rich region',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: newcpgseek58/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+newcpgseek58%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'newcpgseek',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_codcmp.xml',\n",
       "  'description': 'Codon usage table comparison',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: codcmp12/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+codcmp12%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'codcmp',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/wc_gnu.xml',\n",
       "  'description': 'of a dataset',\n",
       "  'id': 'wc_gnu',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=wc_gnu',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Line/Word/Character count',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/drosofff/msp_cap3/ddb463fdf57e/msp_cap3/cap3.xml',\n",
       "  'description': 'Sequence Assembly tool',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/drosofff/msp_cap3/cap3/1.2.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdrosofff%2Fmsp_cap3%2Fcap3%2F1.2.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'cap3',\n",
       "  'panel_section_id': 'ngs:_de_novo',\n",
       "  'panel_section_name': 'NGS: De Novo',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'ddb463fdf57e',\n",
       "   'name': 'msp_cap3',\n",
       "   'owner': 'drosofff',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.2.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/macs2/6d4babad010f/macs2/macs2_predictd.xml',\n",
       "  'description': \"Predict 'd' or fragment size from alignment results\",\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_predictd/2.1.0.20151222.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fmacs2%2Fmacs2_predictd%2F2.1.0.20151222.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MACS2 predictd',\n",
       "  'panel_section_id': 'ngs:_peak_calling',\n",
       "  'panel_section_name': 'NGS: Peak Calling',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '6d4babad010f',\n",
       "   'name': 'macs2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.1.0.20151222.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/gtf2bedgraph.xml',\n",
       "  'description': 'converter',\n",
       "  'id': 'gtf2bedgraph',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=gtf2bedgraph',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'GTF-to-BEDGraph',\n",
       "  'panel_section_id': 'convert',\n",
       "  'panel_section_name': 'Convert Formats',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/unionBedGraphs.xml',\n",
       "  'description': 'combines coverage intervals from multiple BEDGRAPH files',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_unionbedgraph/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_unionbedgraph%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Merge BedGraph files',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_newcpgreport.xml',\n",
       "  'description': 'Report CpG rich areas',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: newcpgreport57/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+newcpgreport57%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'newcpgreport',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_sigcleave.xml',\n",
       "  'description': 'Reports protein signal cleavage sites',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: sigcleave88/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+sigcleave88%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'sigcleave',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/7f3c448e119b/ncbi_blast_plus/tools/ncbi_blast_plus/ncbi_blastdbcmd_info.xml',\n",
       "  'description': 'Show BLAST database information from blastdbcmd',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_blastdbcmd_info/0.1.07',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fncbi_blast_plus%2Fncbi_blastdbcmd_info%2F0.1.07',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'NCBI BLAST+ database info',\n",
       "  'panel_section_id': 'ngs:_mapping',\n",
       "  'panel_section_name': 'NGS: Mapping',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '7f3c448e119b',\n",
       "   'name': 'ncbi_blast_plus',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.1.07'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_union.xml',\n",
       "  'description': 'Reads sequence fragments and builds one sequence',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: union105/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+union105%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'union',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/koen/gbsx/3b325560887b/gbsx/gbsx_digest.xml',\n",
       "  'description': 'simulates a digest for the selection of the best enzymes',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/koen/gbsx/gbsx_digest/1.1.5',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fkoen%2Fgbsx%2Fgbsx_digest%2F1.1.5',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'GBSX: Digest',\n",
       "  'panel_section_id': 'ngs:_rads',\n",
       "  'panel_section_name': 'NGS: RADs',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '3b325560887b',\n",
       "   'name': 'gbsx',\n",
       "   'owner': 'koen',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.1.5'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/gatk2/84584664264c/gatk2/variant_apply_recalibration.xml',\n",
       "  'description': '',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/gatk2/gatk2_variant_apply_recalibration/2.8.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fgatk2%2Fgatk2_variant_apply_recalibration%2F2.8.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Apply Variant Recalibration',\n",
       "  'panel_section_id': 'ngs:_gatk',\n",
       "  'panel_section_name': 'NGS: GATK',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '84584664264c',\n",
       "   'name': 'gatk2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.8.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/complementBed.xml',\n",
       "  'description': 'Extract intervals not represented by an interval file',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_complementbed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_complementbed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'ComplementBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_pepstats.xml',\n",
       "  'description': 'Protein statistics',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: pepstats71/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+pepstats71%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'pepstats',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/maf/interval2maf.xml',\n",
       "  'description': 'given a set of genomic intervals',\n",
       "  'id': 'Interval2Maf1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=Interval2Maf1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Extract MAF blocks',\n",
       "  'panel_section_id': 'fetchAlign',\n",
       "  'panel_section_name': 'Fetch Alignments',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/galaxy-central/tools/plotting/bar_chart.xml',\n",
       "  'description': 'for multiple columns',\n",
       "  'id': 'barchart_gnuplot',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=barchart_gnuplot',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Bar chart',\n",
       "  'panel_section_id': 'plots',\n",
       "  'panel_section_name': 'Graph/Display Data',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/intersectBed.xml',\n",
       "  'description': 'find overlapping intervals in various ways',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_intersectbed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_intersectbed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Intersect intervals',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/gatk2/84584664264c/gatk2/print_reads.xml',\n",
       "  'description': 'on BAM files',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/gatk2/gatk2_print_reads/2.8.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fgatk2%2Fgatk2_print_reads%2F2.8.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Print Reads',\n",
       "  'panel_section_id': 'ngs:_gatk',\n",
       "  'panel_section_name': 'NGS: GATK',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '84584664264c',\n",
       "   'name': 'gatk2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.8.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/wiggle_to_simple_converter.xml',\n",
       "  'description': '',\n",
       "  'id': 'CONVERTER_wiggle_to_interval_0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=CONVERTER_wiggle_to_interval_0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Wiggle to Interval',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/lastz/60afcc2c1d05/lastz/lastz_wrapper.xml',\n",
       "  'description': 'map short reads against reference sequence',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/lastz/lastz_wrapper_2/1.2.2',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Flastz%2Flastz_wrapper_2%2F1.2.2',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Lastz',\n",
       "  'panel_section_id': 'ngs:_mapping',\n",
       "  'panel_section_name': 'NGS: Mapping',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '60afcc2c1d05',\n",
       "   'name': 'lastz',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.2.2'},\n",
       " {'config_file': '/shed_tools/testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/8bb5ea34809d/stacks/STACKS_denovomap.xml',\n",
       "  'description': 'Run the STACKS denovo_map.pl wrapper',\n",
       "  'id': 'testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/STACKSdenovomap/1.2.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=testtoolshed.g2.bx.psu.edu%2Frepos%2Ftiagoantao%2Fstacks%2FSTACKSdenovomap%2F1.2.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'STACKS : De novo map',\n",
       "  'panel_section_id': 'ngs:_rads',\n",
       "  'panel_section_name': 'NGS: RADs',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '8bb5ea34809d',\n",
       "   'name': 'stacks',\n",
       "   'owner': 'tiagoantao',\n",
       "   'tool_shed': 'testtoolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.2.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_skipseq.xml',\n",
       "  'description': 'Reads and writes sequences, skipping first few',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: skipseq91/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+skipseq91%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'skipseq',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_splitter.xml',\n",
       "  'description': 'Split a sequence into (overlapping) smaller sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: splitter92/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+splitter92%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'splitter',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/7f3c448e119b/ncbi_blast_plus/tools/ncbi_blast_plus/ncbi_blastn_wrapper.xml',\n",
       "  'description': 'Search nucleotide database with nucleotide query sequence(s)',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_blastn_wrapper/0.1.07',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fncbi_blast_plus%2Fncbi_blastn_wrapper%2F0.1.07',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'NCBI BLAST+ blastn',\n",
       "  'panel_section_id': 'ngs:_mapping',\n",
       "  'panel_section_name': 'NGS: Mapping',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '7f3c448e119b',\n",
       "   'name': 'ncbi_blast_plus',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.1.07'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/modmine.xml',\n",
       "  'description': 'server',\n",
       "  'id': 'modmine',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=modmine',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'modENCODE modMine',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_MarkDuplicates.xml',\n",
       "  'description': 'examine aligned records in BAM datasets to locate duplicate molecules',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicates/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_MarkDuplicates%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MarkDuplicates',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/bwa/546ada4a9f43/bwa/bwa-mem.xml',\n",
       "  'description': '- map medium and long reads (> 100 bp) against reference genome',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa_mem/0.7.12.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fbwa%2Fbwa_mem%2F0.7.12.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Map with BWA-MEM',\n",
       "  'panel_section_id': 'ngs:_mapping',\n",
       "  'panel_section_name': 'NGS: Mapping',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '546ada4a9f43',\n",
       "   'name': 'bwa',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.7.12.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_tcode.xml',\n",
       "  'description': 'Fickett TESTCODE statistic to identify protein-coding DNA',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: tcode97/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+tcode97%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'tcode',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/data_manager_picard_index_builder/00491eabe22b/data_manager_picard_index_builder/data_manager/picard_index_builder.xml',\n",
       "  'description': 'builder',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/data_manager_picard_index_builder/picard_index_builder_data_manager/0.0.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fdata_manager_picard_index_builder%2Fpicard_index_builder_data_manager%2F0.0.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'DataManagerTool',\n",
       "  'name': 'Picard index',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '00491eabe22b',\n",
       "   'name': 'data_manager_picard_index_builder',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/bed12ToBed6.xml',\n",
       "  'description': 'converter',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bed12tobed6/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_bed12tobed6%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'BED12 to BED6',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_maskseq.xml',\n",
       "  'description': 'Mask off regions of a sequence',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: maskseq51/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+maskseq51%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'maskseq',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_patmatdb.xml',\n",
       "  'description': 'Search a protein sequence with a motif',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: patmatdb67/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+patmatdb67%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'patmatdb',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/maf/vcf_to_maf_customtrack.xml',\n",
       "  'description': 'for display at UCSC',\n",
       "  'id': 'vcf_to_maf_customtrack1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=vcf_to_maf_customtrack1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'VCF to MAF Custom Track',\n",
       "  'panel_section_id': 'plots',\n",
       "  'panel_section_name': 'Graph/Display Data',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_cpgplot.xml',\n",
       "  'description': 'Plot CpG rich areas',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: cpgplot15/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+cpgplot15%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'cpgplot',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_notseq.xml',\n",
       "  'description': 'Exclude a set of sequences and write out the remaining ones',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: notseq61/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+notseq61%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'notseq',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_water.xml',\n",
       "  'description': 'Smith-Waterman local alignment',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: water107/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+water107%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'water',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/genomeCoverageBed.xml',\n",
       "  'description': 'compute the coverage over an entire genome',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_genomecoveragebed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_genomecoveragebed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Genome Coverage',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/tools/imp_exp/imp_history_from_archive.xml',\n",
       "  'description': '',\n",
       "  'id': '__IMPORT_HISTORY__',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=__IMPORT_HISTORY__',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'ImportHistoryTool',\n",
       "  'name': 'Import History',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '0.1'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/interval_to_bigwig_converter.xml',\n",
       "  'description': '',\n",
       "  'id': 'CONVERTER_interval_to_bigwig_0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=CONVERTER_interval_to_bigwig_0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert Genomic Intervals To Coverage',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/8bb5ea34809d/stacks/STACKS_clone_filter.xml',\n",
       "  'description': 'Run the STACKS clone_filter program',\n",
       "  'id': 'testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/STACKS_clone_filter/0.9.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=testtoolshed.g2.bx.psu.edu%2Frepos%2Ftiagoantao%2Fstacks%2FSTACKS_clone_filter%2F0.9.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'STACKS: clone_filter',\n",
       "  'panel_section_id': 'ngs:_rads',\n",
       "  'panel_section_name': 'NGS: RADs',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '8bb5ea34809d',\n",
       "   'name': 'stacks',\n",
       "   'owner': 'tiagoantao',\n",
       "   'tool_shed': 'testtoolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.9.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_trimseq.xml',\n",
       "  'description': 'Trim ambiguous bits off the ends of sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: trimseq103/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+trimseq103%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'trimseq',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/fastqc/28d39af2dd06/fastqc/rgFastQC.xml',\n",
       "  'description': 'Read Quality reports',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/fastqc/fastqc/0.64',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Ffastqc%2Ffastqc%2F0.64',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'FastQC',\n",
       "  'panel_section_id': 'ngs:_qc_and_manipulation',\n",
       "  'panel_section_name': 'NGS: QC and manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '28d39af2dd06',\n",
       "   'name': 'fastqc',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.64'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_DownsampleSam.xml',\n",
       "  'description': 'Downsample a file to retain a subset of the reads',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_DownsampleSam/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_DownsampleSam%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Downsample SAM/BAM',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/cuffquant/986b63735a5e/cuffquant/cuffquant_wrapper.xml',\n",
       "  'description': 'Precompute gene expression levels',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/cuffquant/cuffquant/2.2.1.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fcuffquant%2Fcuffquant%2F2.2.1.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Cuffquant',\n",
       "  'panel_section_id': 'ngs:_rna_analysis',\n",
       "  'panel_section_name': 'NGS: RNA Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '986b63735a5e',\n",
       "   'name': 'cuffquant',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.2.1.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/fasta_to_bowtie_base_index_converter.xml',\n",
       "  'description': '',\n",
       "  'id': 'CONVERTER_fasta_to_bowtie_base_index',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=CONVERTER_fasta_to_bowtie_base_index',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert FASTA to Bowtie base space Index',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/48b5f719e36a/str_fm/readdepth2sequencingdepth.xml',\n",
       "  'description': 'for flank-based mapping of microsatellites',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/readdepth2seqdepth/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Farkarachai-fungtammasan%2Fstr_fm%2Freaddepth2seqdepth%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert informative read depth to sequencing depth',\n",
       "  'panel_section_id': 'str-fm:_microsatellite_analysis',\n",
       "  'panel_section_name': 'STR-FM: Microsatellite Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '48b5f719e36a',\n",
       "   'name': 'str_fm',\n",
       "   'owner': 'arkarachai-fungtammasan',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/data_manager_bwa_index_builder/cc5ae94acf01/data_manager_bwa_index_builder/data_manager/bwa_index_builder.xml',\n",
       "  'description': 'builder',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/data_manager_bwa_index_builder/bwa_index_builder_data_manager/0.0.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fdata_manager_bwa_index_builder%2Fbwa_index_builder_data_manager%2F0.0.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'DataManagerTool',\n",
       "  'name': 'BWA index',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'cc5ae94acf01',\n",
       "   'name': 'data_manager_bwa_index_builder',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.0.1'},\n",
       " {'config_file': '/galaxy-central/tools/maf/maf_reverse_complement.xml',\n",
       "  'description': 'a MAF file',\n",
       "  'id': 'MAF_Reverse_Complement_1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=MAF_Reverse_Complement_1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Reverse Complement',\n",
       "  'panel_section_id': 'fetchAlign',\n",
       "  'panel_section_name': 'Fetch Alignments',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/interval_to_bed_converter.xml',\n",
       "  'description': '',\n",
       "  'id': 'CONVERTER_interval_to_bed_0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=CONVERTER_interval_to_bed_0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert Genomic Intervals To BED',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/fastq_paired_end_joiner/6a7f5da7c76d/fastq_paired_end_joiner/fastq_paired_end_joiner.xml',\n",
       "  'description': 'on paired end reads',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/fastq_paired_end_joiner/fastq_paired_end_joiner/2.0.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Ffastq_paired_end_joiner%2Ffastq_paired_end_joiner%2F2.0.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'FASTQ joiner',\n",
       "  'panel_section_id': 'ngs:_qc_and_manipulation',\n",
       "  'panel_section_name': 'NGS: QC and manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '6a7f5da7c76d',\n",
       "   'name': 'fastq_paired_end_joiner',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_CollectWgsMetrics.xml',\n",
       "  'description': 'compute metrics for evaluating of whole genome sequencing experiments',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_CollectWgsMetrics/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_CollectWgsMetrics%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'CollectWgsMetrics',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/sorter.xml',\n",
       "  'description': 'data in ascending or descending order',\n",
       "  'id': 'sort1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=sort1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Sort',\n",
       "  'panel_section_id': 'filter',\n",
       "  'panel_section_name': 'Filter and Sort',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.3'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/snpeff/e09ce114d240/snpeff/snpEff_download.xml',\n",
       "  'description': 'Download a new database',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/snpeff/snpEff_download/4.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsnpeff%2FsnpEff_download%2F4.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'SnpEff Download',\n",
       "  'panel_section_id': 'ngs:_variant_analysis',\n",
       "  'panel_section_name': 'NGS: Variant Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'e09ce114d240',\n",
       "   'name': 'snpeff',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '4.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/data_manager_fetch_genome_all_fasta/fb744a070bee/data_manager_fetch_genome_all_fasta/data_manager/data_manager_fetch_genome_all_fasta.xml',\n",
       "  'description': 'fetching',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/data_manager_fetch_genome_all_fasta/data_manager_fetch_genome_all_fasta/0.0.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fdata_manager_fetch_genome_all_fasta%2Fdata_manager_fetch_genome_all_fasta%2F0.0.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'DataManagerTool',\n",
       "  'name': 'Reference Genome',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'fb744a070bee',\n",
       "   'name': 'data_manager_fetch_genome_all_fasta',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/cuffmerge/8160c8ea4eb9/cuffmerge/cuffmerge_wrapper.xml',\n",
       "  'description': 'merge together several Cufflinks assemblies',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/cuffmerge/cuffmerge/2.2.1.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fcuffmerge%2Fcuffmerge%2F2.2.1.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Cuffmerge',\n",
       "  'panel_section_id': 'ngs:_rna_analysis',\n",
       "  'panel_section_name': 'NGS: RNA Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '8160c8ea4eb9',\n",
       "   'name': 'cuffmerge',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.2.1.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_syco.xml',\n",
       "  'description': 'Synonymous codon usage Gribskov statistic plot',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: syco96/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+syco96%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'syco',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_compseq.xml',\n",
       "  'description': 'Count composition of dimer/trimer/etc words in a sequence',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: compseq14/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+compseq14%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'compseq',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_merger.xml',\n",
       "  'description': 'Merge two overlapping nucleic acid sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: merger54/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+merger54%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'merger',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/macs2/6d4babad010f/macs2/macs2_bdgpeakcall.xml',\n",
       "  'description': 'Call peaks from bedGraph output',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_bdgpeakcall/2.1.0.20151222.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fmacs2%2Fmacs2_bdgpeakcall%2F2.1.0.20151222.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MACS2 bdgpeakcall',\n",
       "  'panel_section_id': 'ngs:_peak_calling',\n",
       "  'panel_section_name': 'NGS: Peak Calling',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '6d4babad010f',\n",
       "   'name': 'macs2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.1.0.20151222.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_dottup.xml',\n",
       "  'description': 'Displays a wordmatch dotplot of two sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: dottup26/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+dottup26%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'dottup',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_pepnet.xml',\n",
       "  'description': 'Displays proteins as a helical net',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: pepnet70/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+pepnet70%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'pepnet',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_BedToIntervalList.xml',\n",
       "  'description': 'convert coordinate data into picard interval list format',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_BedToIntervalList/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_BedToIntervalList%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'BedToIntervalList',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/cutWrapper.xml',\n",
       "  'description': 'columns from a table',\n",
       "  'id': 'Cut1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=Cut1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Cut',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.2'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_prettyplot.xml',\n",
       "  'description': 'Displays aligned sequences, with colouring and boxing',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: prettyplot79/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+prettyplot79%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'prettyplot',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/data_manager_fetch_genome_dbkeys_all_fasta/776bb1b478a0/data_manager_fetch_genome_dbkeys_all_fasta/data_manager/data_manager_fetch_genome_all_fasta_dbkeys.xml',\n",
       "  'description': 'fetching',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/data_manager_fetch_genome_dbkeys_all_fasta/data_manager_fetch_genome_all_fasta_dbkey/0.0.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fdata_manager_fetch_genome_dbkeys_all_fasta%2Fdata_manager_fetch_genome_all_fasta_dbkey%2F0.0.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'DataManagerTool',\n",
       "  'name': 'Create DBKey and Reference Genome',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '776bb1b478a0',\n",
       "   'name': 'data_manager_fetch_genome_dbkeys_all_fasta',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_ReorderSam.xml',\n",
       "  'description': 'reorder reads to match ordering in reference sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_ReorderSam/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_ReorderSam%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'ReorderSam',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/multiIntersectBed.xml',\n",
       "  'description': 'identifies common intervals among multiple interval files',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multiintersectbed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_multiintersectbed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Multiple Intersect',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/data_manager_build_kraken_database/c3276afba416/data_manager_build_kraken_database/data_manager/kraken_database_builder.xml',\n",
       "  'description': 'database builder',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/data_manager_build_kraken_database/kraken_database_builder/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fdata_manager_build_kraken_database%2Fkraken_database_builder%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'DataManagerTool',\n",
       "  'name': 'Kraken',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'c3276afba416',\n",
       "   'name': 'data_manager_build_kraken_database',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/7f3c448e119b/ncbi_blast_plus/tools/ncbi_blast_plus/ncbi_segmasker_wrapper.xml',\n",
       "  'description': 'low-complexity regions in protein sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_segmasker_wrapper/0.1.07',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fncbi_blast_plus%2Fncbi_segmasker_wrapper%2F0.1.07',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'NCBI BLAST+ segmasker',\n",
       "  'panel_section_id': 'ngs:_mapping',\n",
       "  'panel_section_name': 'NGS: Mapping',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '7f3c448e119b',\n",
       "   'name': 'ncbi_blast_plus',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.1.07'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/gatk2/84584664264c/gatk2/base_recalibrator.xml',\n",
       "  'description': 'calculates covariates used to recalibrate base quality scores of reads',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/gatk2/gatk2_base_recalibrator/2.8.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fgatk2%2Fgatk2_base_recalibrator%2F2.8.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Base Recalibrator',\n",
       "  'panel_section_id': 'ngs:_gatk',\n",
       "  'panel_section_name': 'NGS: GATK',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '84584664264c',\n",
       "   'name': 'gatk2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.8.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_shuffleseq.xml',\n",
       "  'description': 'Shuffles a set of sequences maintaining composition',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: shuffleseq87/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+shuffleseq87%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'shuffleseq',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/7f3c448e119b/ncbi_blast_plus/tools/ncbi_blast_plus/ncbi_tblastn_wrapper.xml',\n",
       "  'description': 'Search translated nucleotide database with protein query sequence(s)',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_tblastn_wrapper/0.1.07',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fncbi_blast_plus%2Fncbi_tblastn_wrapper%2F0.1.07',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'NCBI BLAST+ tblastn',\n",
       "  'panel_section_id': 'ngs:_mapping',\n",
       "  'panel_section_name': 'NGS: Mapping',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '7f3c448e119b',\n",
       "   'name': 'ncbi_blast_plus',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.1.07'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/vcf2tsv/d6c944f1581a/vcf2tsv/vcf2tsv.xml',\n",
       "  'description': 'Convert VCF data into TAB-delimited format',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/vcf2tsv/vcf2tsv/0.0.3',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fvcf2tsv%2Fvcf2tsv%2F0.0.3',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'VCFtoTab-delimited:',\n",
       "  'panel_section_id': 'ngs:_vcf_manipulation',\n",
       "  'panel_section_name': 'NGS: VCF Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'd6c944f1581a',\n",
       "   'name': 'vcf2tsv',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.0.3'},\n",
       " {'config_file': '/shed_tools/testtoolshed.g2.bx.psu.edu/repos/tiagoantao/barcode_stacks/2e309e9cacba/barcode_stacks/barcode_sort.xml',\n",
       "  'description': 'Barcode Sorting with Paired-End reads for Stacks',\n",
       "  'id': 'testtoolshed.g2.bx.psu.edu/repos/tiagoantao/barcode_stacks/barcode_sort_stacks/2.0.2',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=testtoolshed.g2.bx.psu.edu%2Frepos%2Ftiagoantao%2Fbarcode_stacks%2Fbarcode_sort_stacks%2F2.0.2',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Barcode sorting for Stacks',\n",
       "  'panel_section_id': 'ngs:_rads',\n",
       "  'panel_section_name': 'NGS: RADs',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '2e309e9cacba',\n",
       "   'name': 'barcode_stacks',\n",
       "   'owner': 'tiagoantao',\n",
       "   'tool_shed': 'testtoolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.0.2'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/multiCov.xml',\n",
       "  'description': 'counts coverage from multiple BAMs at specific intervals',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_multicovtbed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_multicovtbed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MultiCovBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/galaxy-central/tools/stats/gsummary.xml',\n",
       "  'description': 'for any numerical column',\n",
       "  'id': 'Summary_Statistics1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=Summary_Statistics1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Summary Statistics',\n",
       "  'panel_section_id': 'stats',\n",
       "  'panel_section_name': 'Statistics',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.1.0'},\n",
       " {'config_file': '/galaxy-central/tools/plotting/boxplot.xml',\n",
       "  'description': 'of quality statistics',\n",
       "  'id': 'qual_stats_boxplot',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=qual_stats_boxplot',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Boxplot',\n",
       "  'panel_section_id': 'plots',\n",
       "  'panel_section_name': 'Graph/Display Data',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_cusp.xml',\n",
       "  'description': 'Create a codon usage table',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: cusp17/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+cusp17%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'cusp',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_matcher.xml',\n",
       "  'description': 'Finds the best local alignments between two sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: matcher52/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+matcher52%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'matcher',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/nick/allele_counts/a72277535a2c/allele_counts/allele-counts.xml',\n",
       "  'description': 'process variant counts',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/nick/allele_counts/allele_counts_1/1.2',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fnick%2Fallele_counts%2Fallele_counts_1%2F1.2',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Variant Annotator',\n",
       "  'panel_section_id': 'ngs:_variant_analysis',\n",
       "  'panel_section_name': 'NGS: Variant Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'a72277535a2c',\n",
       "   'name': 'allele_counts',\n",
       "   'owner': 'nick',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.2'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_MergeSamFiles.xml',\n",
       "  'description': 'merges multiple SAM/BAM datasets into one',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MergeSamFiles/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_MergeSamFiles%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MergeSamFiles',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_hmoment.xml',\n",
       "  'description': 'Hydrophobic moment calculation',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: hmoment44/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+hmoment44%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'hmoment',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_MeanQualityByCycle.xml',\n",
       "  'description': 'chart distribution of base qualities',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MeanQualityByCycle/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_MeanQualityByCycle%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MeanQualityByCycle',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/bed_to_bigbed.xml',\n",
       "  'description': 'converter',\n",
       "  'id': 'bed_to_bigBed',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=bed_to_bigBed',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'BED-to-bigBed',\n",
       "  'panel_section_id': 'convert',\n",
       "  'panel_section_name': 'Convert Formats',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/gatk2/84584664264c/gatk2/variant_annotator.xml',\n",
       "  'description': '',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/gatk2/gatk2_variant_annotator/2.8.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fgatk2%2Fgatk2_variant_annotator%2F2.8.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Variant Annotator',\n",
       "  'panel_section_id': 'ngs:_gatk',\n",
       "  'panel_section_name': 'NGS: GATK',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '84584664264c',\n",
       "   'name': 'gatk2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.8.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_tmap.xml',\n",
       "  'description': 'Displays membrane spanning regions',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: tmap99/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+tmap99%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'tmap',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/gramene_mart.xml',\n",
       "  'description': 'Central server',\n",
       "  'id': 'gramenemart',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=gramenemart',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'GrameneMart',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/slopBed.xml',\n",
       "  'description': 'adjust the size of intervals',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_slopbed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_slopbed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'SlopBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/hbvar.xml',\n",
       "  'description': 'Human Hemoglobin Variants and Thalassemias',\n",
       "  'id': 'hbvar',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=hbvar',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'HbVar',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '2.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/48b5f719e36a/str_fm/combineprobforallelecombination.xml',\n",
       "  'description': 'from the same allele combination',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/combineproballelecom/2.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Farkarachai-fungtammasan%2Fstr_fm%2Fcombineproballelecom%2F2.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Combine read profile probabilities ',\n",
       "  'panel_section_id': 'str-fm:_microsatellite_analysis',\n",
       "  'panel_section_name': 'STR-FM: Microsatellite Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '48b5f719e36a',\n",
       "   'name': 'str_fm',\n",
       "   'owner': 'arkarachai-fungtammasan',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.0.0'},\n",
       " {'config_file': '/shed_tools/testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/8bb5ea34809d/stacks/STACKS_refmap.xml',\n",
       "  'description': 'Run the STACKS ref_map.pl wrapper',\n",
       "  'id': 'testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/STACKSrefmap/1.2.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=testtoolshed.g2.bx.psu.edu%2Frepos%2Ftiagoantao%2Fstacks%2FSTACKSrefmap%2F1.2.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'STACKS : Reference map',\n",
       "  'panel_section_id': 'ngs:_rads',\n",
       "  'panel_section_name': 'NGS: RADs',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '8bb5ea34809d',\n",
       "   'name': 'stacks',\n",
       "   'owner': 'tiagoantao',\n",
       "   'tool_shed': 'testtoolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.2.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/dgidb_annotator/c5bb987015c5/dgidb_annotator/dgidb_annotator.xml',\n",
       "  'description': 'database info',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/dgidb_annotator/dgidb_annotate/0.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fdgidb_annotator%2Fdgidb_annotate%2F0.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Annotate with DGI',\n",
       "  'panel_section_id': 'phenotype_association',\n",
       "  'panel_section_name': 'Phenotype Association',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'c5bb987015c5',\n",
       "   'name': 'dgidb_annotator',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_MarkDuplicatesWithMateCigar.xml',\n",
       "  'description': 'examine aligned records in BAM datasets to locate duplicate molecules',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_MarkDuplicatesWithMateCigar/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_MarkDuplicatesWithMateCigar%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MarkDuplicatesWithMateCigar',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/fastq_groomer/e4d28c94242d/fastq_groomer/fastq_groomer.xml',\n",
       "  'description': 'convert between various FASTQ quality formats',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/fastq_groomer/fastq_groomer/1.0.4',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Ffastq_groomer%2Ffastq_groomer%2F1.0.4',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'FASTQ Groomer',\n",
       "  'panel_section_id': 'ngs:_qc_and_manipulation',\n",
       "  'panel_section_name': 'NGS: QC and manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'e4d28c94242d',\n",
       "   'name': 'fastq_groomer',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.4'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/288462ec2630/text_processing/head.xml',\n",
       "  'description': 'lines from a dataset (head)',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_head_tool/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Ftext_processing%2Ftp_head_tool%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Select first',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '288462ec2630',\n",
       "   'name': 'text_processing',\n",
       "   'owner': 'bgruening',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_epestfind.xml',\n",
       "  'description': 'Finds PEST motifs as potential proteolytic cleavage sites',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: epestfind29/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+epestfind29%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'epestfind',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_CleanSam.xml',\n",
       "  'description': 'perform SAM/BAM grooming',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_CleanSam/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_CleanSam%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'CleanSam',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_biosed.xml',\n",
       "  'description': 'Replace or delete sequence sections',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: biosed4/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+biosed4%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'biosed',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/cuffdiff/0f32ad418df8/cuffdiff/cuffdiff_wrapper.xml',\n",
       "  'description': 'find significant changes in transcript expression, splicing, and promoter use',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/cuffdiff/cuffdiff/2.2.1.3',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fcuffdiff%2Fcuffdiff%2F2.2.1.3',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Cuffdiff',\n",
       "  'panel_section_id': 'ngs:_rna_analysis',\n",
       "  'panel_section_name': 'NGS: RNA Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '0f32ad418df8',\n",
       "   'name': 'cuffdiff',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.2.1.3'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/snpeff/e09ce114d240/snpeff/snpEff.xml',\n",
       "  'description': 'Variant effect and annotation',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/snpeff/snpEff/4.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsnpeff%2FsnpEff%2F4.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'SnpEff',\n",
       "  'panel_section_id': 'ngs:_variant_analysis',\n",
       "  'panel_section_name': 'NGS: Variant Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'e09ce114d240',\n",
       "   'name': 'snpeff',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '4.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_antigenic.xml',\n",
       "  'description': 'Predicts potentially antigenic regions of a protein sequence, using the method of Kolaskar and Tongaonkar.',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: antigenic1/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+antigenic1%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'antigenic',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/fastq_trimmer/e0cfb5a703ce/fastq_trimmer/fastq_trimmer.xml',\n",
       "  'description': 'by column',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/fastq_trimmer/fastq_trimmer/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Ffastq_trimmer%2Ffastq_trimmer%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'FASTQ Trimmer',\n",
       "  'panel_section_id': 'ngs:_qc_and_manipulation',\n",
       "  'panel_section_name': 'NGS: QC and manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'e0cfb5a703ce',\n",
       "   'name': 'fastq_trimmer',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_fuzznuc.xml',\n",
       "  'description': 'Nucleic acid pattern search',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: fuzznuc37/5.0.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+fuzznuc37%2F5.0.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'fuzznuc',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/vcfgeno2haplo/401f29f278f0/vcfgeno2haplo/vcfgeno2haplo.xml',\n",
       "  'description': 'Convert genotype-based phased alleles into haplotype alleles',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/vcfgeno2haplo/vcfgeno2haplo/0.0.3',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fvcfgeno2haplo%2Fvcfgeno2haplo%2F0.0.3',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'VCFgenotype-to-haplotype:',\n",
       "  'panel_section_id': 'ngs:_vcf_manipulation',\n",
       "  'panel_section_name': 'NGS: VCF Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '401f29f278f0',\n",
       "   'name': 'vcfgeno2haplo',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.0.3'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/tophat2/4eb3c3beb9c7/tophat2/tophat2_wrapper.xml',\n",
       "  'description': 'Gapped-read mapper for RNA-seq data',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/tophat2/tophat2/0.9',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Ftophat2%2Ftophat2%2F0.9',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'TopHat',\n",
       "  'panel_section_id': 'ngs:_rna_analysis',\n",
       "  'panel_section_name': 'NGS: RNA Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '4eb3c3beb9c7',\n",
       "   'name': 'tophat2',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.9'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/canonical_correlation_analysis/9bc0c48a027f/canonical_correlation_analysis/cca.xml',\n",
       "  'description': '',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/canonical_correlation_analysis/cca1/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fcanonical_correlation_analysis%2Fcca1%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Canonical Correlation Analysis',\n",
       "  'panel_section_id': '',\n",
       "  'panel_section_name': '',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '9bc0c48a027f',\n",
       "   'name': 'canonical_correlation_analysis',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/bwa/546ada4a9f43/bwa/bwa.xml',\n",
       "  'description': '- map short reads (< 100 bp) against reference genome',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/bwa/bwa/0.7.12.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fbwa%2Fbwa%2F0.7.12.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Map with BWA',\n",
       "  'panel_section_id': 'ngs:_mapping',\n",
       "  'panel_section_name': 'NGS: Mapping',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '546ada4a9f43',\n",
       "   'name': 'bwa',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.7.12.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_needle.xml',\n",
       "  'description': 'Needleman-Wunsch global alignment',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: needle56/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+needle56%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'needle',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/interval_to_bed6_converter.xml',\n",
       "  'description': '',\n",
       "  'id': 'CONVERTER_interval_to_bed6_0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=CONVERTER_interval_to_bed6_0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert Genomic Intervals To Strict BED6',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/fastx_clipper/bdf1ce174e39/fastx_clipper/fastx_clipper.xml',\n",
       "  'description': 'adapter sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/fastx_clipper/cshl_fastx_clipper/1.0.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Ffastx_clipper%2Fcshl_fastx_clipper%2F1.0.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Clip',\n",
       "  'panel_section_id': 'ngs:_qc_and_manipulation',\n",
       "  'panel_section_name': 'NGS: QC and manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'bdf1ce174e39',\n",
       "   'name': 'fastx_clipper',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/galaxy-central/tools/filters/wig_to_bigwig.xml',\n",
       "  'description': 'converter',\n",
       "  'id': 'wig_to_bigWig',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=wig_to_bigWig',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Wig/BedGraph-to-bigWig',\n",
       "  'panel_section_id': 'convert',\n",
       "  'panel_section_name': 'Convert Formats',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.1.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/gff/extract_GFF_Features.xml',\n",
       "  'description': 'from GFF data',\n",
       "  'id': 'Extract_features1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=Extract_features1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Extract features',\n",
       "  'panel_section_id': 'filter',\n",
       "  'panel_section_name': 'Filter and Sort',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_extractseq.xml',\n",
       "  'description': 'Extract regions from a sequence',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: extractseq35/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+extractseq35%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'extractseq',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/pbed_ldreduced_converter.xml',\n",
       "  'description': '',\n",
       "  'id': 'pbed2ldindepconvert',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=pbed2ldindepconvert',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert plink pbed to ld reduced format',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '0.01'},\n",
       " {'config_file': '/galaxy-central/tools/filters/randomlines.xml',\n",
       "  'description': 'from a file',\n",
       "  'id': 'random_lines1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=random_lines1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Select random lines',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '2.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_cutseq.xml',\n",
       "  'description': 'Removes a specified section from a sequence',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: cutseq18/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+cutseq18%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'cutseq',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/bamToBed.xml',\n",
       "  'description': 'converter',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_bamtobed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_bamtobed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'BAM to BED',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/annotateBed.xml',\n",
       "  'description': 'annotate coverage of features from multiple files',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_annotatebed/2.24.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_annotatebed%2F2.24.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'AnnotateBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.1'},\n",
       " {'config_file': '/shed_tools/testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/8bb5ea34809d/stacks/STACKS_population.xml',\n",
       "  'description': 'Run the STACKS populations program',\n",
       "  'id': 'testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/STACKSpopulation/1.2.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=testtoolshed.g2.bx.psu.edu%2Frepos%2Ftiagoantao%2Fstacks%2FSTACKSpopulation%2F1.2.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'STACKS : populations',\n",
       "  'panel_section_id': 'ngs:_rads',\n",
       "  'panel_section_name': 'NGS: RADs',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '8bb5ea34809d',\n",
       "   'name': 'stacks',\n",
       "   'owner': 'tiagoantao',\n",
       "   'tool_shed': 'testtoolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.2.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/changeCase.xml',\n",
       "  'description': 'of selected columns',\n",
       "  'id': 'ChangeCase',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=ChangeCase',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Change Case',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/b723c120161a/sra_tools/sam_dump.xml',\n",
       "  'description': 'in SAM or BAM format from NCBI SRA.',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/sam_dump/1.2.5',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsra_tools%2Fsam_dump%2F1.2.5',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Extract reads',\n",
       "  'panel_section_id': 'ncbi_sra_tools',\n",
       "  'panel_section_name': 'NCBI SRA Tools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b723c120161a',\n",
       "   'name': 'sra_tools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.2.5'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_RevertSam.xml',\n",
       "  'description': 'revert SAM/BAM datasets to a previous state',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_RevertSam/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_RevertSam%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'RevertSam',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/len_to_linecount.xml',\n",
       "  'description': '',\n",
       "  'id': 'CONVERTER_len_to_linecount',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=CONVERTER_len_to_linecount',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert Len file to Linecount',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/48b5f719e36a/str_fm/profilegenerator.xml',\n",
       "  'description': 'of the consecutive allele from given error profile',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/Profilegenerator/2.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Farkarachai-fungtammasan%2Fstr_fm%2FProfilegenerator%2F2.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Generate all possible combination of STR length profile',\n",
       "  'panel_section_id': 'str-fm:_microsatellite_analysis',\n",
       "  'panel_section_name': 'STR-FM: Microsatellite Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '48b5f719e36a',\n",
       "   'name': 'str_fm',\n",
       "   'owner': 'arkarachai-fungtammasan',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/data_manager_snpeff/9ac823a8b328/data_manager_snpeff/data_manager/data_manager_snpEff_download.xml',\n",
       "  'description': 'Download a new database',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/data_manager_snpeff/data_manager_snpeff_download/4.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fdata_manager_snpeff%2Fdata_manager_snpeff_download%2F4.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'DataManagerTool',\n",
       "  'name': 'SnpEff Download',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '9ac823a8b328',\n",
       "   'name': 'data_manager_snpeff',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '4.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/gff/gff_filter_by_feature_count.xml',\n",
       "  'description': 'using simple expressions',\n",
       "  'id': 'gff_filter_by_feature_count',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=gff_filter_by_feature_count',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Filter GFF data by feature count',\n",
       "  'panel_section_id': 'filter',\n",
       "  'panel_section_name': 'Filter and Sort',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/gatk2/84584664264c/gatk2/depth_of_coverage.xml',\n",
       "  'description': 'on BAM files',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/gatk2/gatk2_depth_of_coverage/2.8.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fgatk2%2Fgatk2_depth_of_coverage%2F2.8.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Depth of Coverage',\n",
       "  'panel_section_id': 'ngs:_gatk',\n",
       "  'panel_section_name': 'NGS: GATK',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '84584664264c',\n",
       "   'name': 'gatk2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.8.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/mutate_snp_codon/8f0af7251167/mutate_snp_codon/mutate_snp_codon.xml',\n",
       "  'description': 'with SNPs',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/mutate_snp_codon/mutate_snp_codon_1/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fmutate_snp_codon%2Fmutate_snp_codon_1%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Mutate Codons',\n",
       "  'panel_section_id': 'ngs:_variant_analysis',\n",
       "  'panel_section_name': 'NGS: Variant Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '8f0af7251167',\n",
       "   'name': 'mutate_snp_codon',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/nucBed.xml',\n",
       "  'description': 'profile the nucleotide content of intervals in a FASTA file',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_nucbed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_nucbed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'NucBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/galaxy-central/tools/genomespace/genomespace_exporter.xml',\n",
       "  'description': '- send data to GenomeSpace',\n",
       "  'id': 'genomespace_exporter',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=genomespace_exporter',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'GenomeSpace Exporter',\n",
       "  'panel_section_id': 'send',\n",
       "  'panel_section_name': 'Send Data',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '0.0.4'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_ValidateSamFile.xml',\n",
       "  'description': 'assess validity of SAM/BAM dataset',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_ValidateSamFile/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_ValidateSamFile%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'ValidateSamFile',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/axt_to_concat_fasta.xml',\n",
       "  'description': 'Converts an AXT formatted file to a concatenated FASTA alignment',\n",
       "  'id': 'axt_to_concat_fasta',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=axt_to_concat_fasta',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'AXT to concatenated FASTA',\n",
       "  'panel_section_id': 'convert',\n",
       "  'panel_section_name': 'Convert Formats',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/tabular_to_dbnsfp.xml',\n",
       "  'description': '',\n",
       "  'id': 'tabular_to_dbnsfp',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=tabular_to_dbnsfp',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert tabular to dbnsfp',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_FixMateInformation.xml',\n",
       "  'description': \"ensure that all mate-pair information is in sync between each read and it's mate pair\",\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_FixMateInformation/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_FixMateInformation%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'FixMateInformation',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/galaxy-central/tools/maf/interval2maf_pairwise.xml',\n",
       "  'description': 'given a set of genomic intervals',\n",
       "  'id': 'Interval2Maf_pairwise1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=Interval2Maf_pairwise1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Extract Pairwise MAF blocks',\n",
       "  'panel_section_id': 'fetchAlign',\n",
       "  'panel_section_name': 'Fetch Alignments',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/gff_to_bed_converter.xml',\n",
       "  'description': '',\n",
       "  'id': 'CONVERTER_gff_to_bed_0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=CONVERTER_gff_to_bed_0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert GFF to BED',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/maf/maf_by_block_number.xml',\n",
       "  'description': 'given a set of block numbers and a MAF file',\n",
       "  'id': 'maf_by_block_number1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=maf_by_block_number1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Extract MAF by block number',\n",
       "  'panel_section_id': 'fetchAlign',\n",
       "  'panel_section_name': 'Fetch Alignments',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/gatk2/84584664264c/gatk2/variant_filtration.xml',\n",
       "  'description': 'on VCF files',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/gatk2/gatk2_variant_filtration/2.8.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fgatk2%2Fgatk2_variant_filtration%2F2.8.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Variant Filtration',\n",
       "  'panel_section_id': 'ngs:_gatk',\n",
       "  'panel_section_name': 'NGS: GATK',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '84584664264c',\n",
       "   'name': 'gatk2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.8.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_iep.xml',\n",
       "  'description': 'Calculates the isoelectric point of a protein',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: iep45/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+iep45%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'iep',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/tools/imp_exp/exp_history_to_archive.xml',\n",
       "  'description': '',\n",
       "  'id': '__EXPORT_HISTORY__',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=__EXPORT_HISTORY__',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'ExportHistoryTool',\n",
       "  'name': 'Export History',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/mapBed.xml',\n",
       "  'description': 'apply a function to a column for each overlapping interval',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_map/2.24.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_map%2F2.24.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MapBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.1'},\n",
       " {'config_file': '/shed_tools/testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/8bb5ea34809d/stacks/bwa_wrapper.xml',\n",
       "  'description': 'from zip file with fastqsanger files',\n",
       "  'id': 'testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/bwa_wrapper_stacks/1.2.3',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=testtoolshed.g2.bx.psu.edu%2Frepos%2Ftiagoantao%2Fstacks%2Fbwa_wrapper_stacks%2F1.2.3',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Map with BWA for STACKS',\n",
       "  'panel_section_id': 'ngs:_rads',\n",
       "  'panel_section_name': 'NGS: RADs',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '8bb5ea34809d',\n",
       "   'name': 'stacks',\n",
       "   'owner': 'tiagoantao',\n",
       "   'tool_shed': 'testtoolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.2.3'},\n",
       " {'config_file': '/galaxy-central/tools/maf/maf_thread_for_species.xml',\n",
       "  'description': 'by Species',\n",
       "  'id': 'MAF_Thread_For_Species1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=MAF_Thread_For_Species1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Join MAF blocks',\n",
       "  'panel_section_id': 'fetchAlign',\n",
       "  'panel_section_name': 'Fetch Alignments',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/288462ec2630/text_processing/multijoin.xml',\n",
       "  'description': '(combine multiple files)',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_multijoin_tool/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Ftext_processing%2Ftp_multijoin_tool%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Multi-Join',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '288462ec2630',\n",
       "   'name': 'text_processing',\n",
       "   'owner': 'bgruening',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/48b5f719e36a/str_fm/fetchflank.xml',\n",
       "  'description': 'the STRs in the reads and output two fastq files in forward-forward orientation',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/fetchflank/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Farkarachai-fungtammasan%2Fstr_fm%2Ffetchflank%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Fetch bases flanking',\n",
       "  'panel_section_id': 'str-fm:_microsatellite_analysis',\n",
       "  'panel_section_name': 'STR-FM: Microsatellite Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '48b5f719e36a',\n",
       "   'name': 'str_fm',\n",
       "   'owner': 'arkarachai-fungtammasan',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/pbed_to_lped_converter.xml',\n",
       "  'description': '',\n",
       "  'id': 'pbed2lpedconvert',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=pbed2lpedconvert',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert plink pbed to linkage lped',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '0.01'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_pepwheel.xml',\n",
       "  'description': 'Shows protein sequences as helices',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: pepwheel72/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+pepwheel72%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'pepwheel',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/set_metadata_tool.xml',\n",
       "  'description': '',\n",
       "  'id': '__SET_METADATA__',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=__SET_METADATA__',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'SetMetadataTool',\n",
       "  'name': 'Set External Metadata',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/vcf_filter/da1a6f33b504/vcf_filter/filter.xml',\n",
       "  'description': 'a VCF file',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/vcf_filter/vcf_filter/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fvcf_filter%2Fvcf_filter%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Filter',\n",
       "  'panel_section_id': 'ngs:_variant_analysis',\n",
       "  'panel_section_name': 'NGS: Variant Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'da1a6f33b504',\n",
       "   'name': 'vcf_filter',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/biomart.xml',\n",
       "  'description': 'Ensembl server',\n",
       "  'id': 'biomart',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=biomart',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'BioMart',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/flankBed.xml',\n",
       "  'description': 'create new intervals from the flanks of existing intervals',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_flankbed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_flankbed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'FlankBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_megamerger.xml',\n",
       "  'description': 'Merge two large overlapping nucleic acid sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: megamerger53/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+megamerger53%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'megamerger',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/sff_extractor.xml',\n",
       "  'description': '',\n",
       "  'id': 'Sff_extractor',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=Sff_extractor',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'SFF converter',\n",
       "  'panel_section_id': 'convert',\n",
       "  'panel_section_name': 'Convert Formats',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_lindna.xml',\n",
       "  'description': 'Draws linear maps of DNA constructs',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: lindna48/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+lindna48%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'lindna',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/spacingBed.xml',\n",
       "  'description': 'reports the distances between features',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_spacingbed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_spacingbed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'SpacingBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_palindrome.xml',\n",
       "  'description': 'Looks for inverted repeats in a nucleotide sequence',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: palindrome65/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+palindrome65%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'palindrome',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/fastqtofasta/723b5b38d88a/fastqtofasta/fastq_to_fasta.xml',\n",
       "  'description': 'converter',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/fastqtofasta/fastq_to_fasta_python/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Ffastqtofasta%2Ffastq_to_fasta_python%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'FASTQ to FASTA',\n",
       "  'panel_section_id': 'ngs:_qc_and_manipulation',\n",
       "  'panel_section_name': 'NGS: QC and manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '723b5b38d88a',\n",
       "   'name': 'fastqtofasta',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_CollectRnaSeqMetrics.xml',\n",
       "  'description': 'collect metrics about the alignment of RNA to various functional classes of loci in the genome',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_CollectRnaSeqMetrics/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_CollectRnaSeqMetrics%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'CollectRnaSeqMetrics',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_plotorf.xml',\n",
       "  'description': 'Plot potential open reading frames',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: plotorf76/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+plotorf76%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'plotorf',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/bgruening/trim_galore/f1e71aeaa923/trim_galore/trim_galore.xml',\n",
       "  'description': 'adaptive quality and adapter trimmer',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/bgruening/trim_galore/trim_galore/0.4.2',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Ftrim_galore%2Ftrim_galore%2F0.4.2',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Trim Galore!',\n",
       "  'panel_section_id': 'ngs:_qc_and_manipulation',\n",
       "  'panel_section_name': 'NGS: QC and manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f1e71aeaa923',\n",
       "   'name': 'trim_galore',\n",
       "   'owner': 'bgruening',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.4.2'},\n",
       " {'config_file': '/galaxy-central/tools/filters/mergeCols.xml',\n",
       "  'description': 'together',\n",
       "  'id': 'mergeCols1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=mergeCols1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Merge Columns',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/7f3c448e119b/ncbi_blast_plus/tools/ncbi_blast_plus/ncbi_dustmasker_wrapper.xml',\n",
       "  'description': 'masks low complexity regions',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_dustmasker_wrapper/0.1.07',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fncbi_blast_plus%2Fncbi_dustmasker_wrapper%2F0.1.07',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'NCBI BLAST+ dustmasker',\n",
       "  'panel_section_id': 'ngs:_mapping',\n",
       "  'panel_section_name': 'NGS: Mapping',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '7f3c448e119b',\n",
       "   'name': 'ncbi_blast_plus',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.1.07'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_etandem.xml',\n",
       "  'description': 'Looks for tandem repeats in a nucleotide sequence',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: etandem33/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+etandem33%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'etandem',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/cmonjeau/ziptool/e466f86a4520/ziptool/zip.xml',\n",
       "  'description': 'and name files',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/cmonjeau/ziptool/zip/1.0.1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fcmonjeau%2Fziptool%2Fzip%2F1.0.1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Zip',\n",
       "  'panel_section_id': 'convert',\n",
       "  'panel_section_name': 'Convert Formats',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'e466f86a4520',\n",
       "   'name': 'ziptool',\n",
       "   'owner': 'cmonjeau',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/closestBed.xml',\n",
       "  'description': 'find the closest, potentially non-overlapping interval',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_closestbed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_closestbed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'ClosestBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/galaxy-central/tools/extract/liftOver_wrapper.xml',\n",
       "  'description': 'between assemblies and genomes',\n",
       "  'id': 'liftOver1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=liftOver1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert genome coordinates',\n",
       "  'panel_section_id': 'liftOver',\n",
       "  'panel_section_name': 'Lift-Over',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.4'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/gatk2/84584664264c/gatk2/unified_genotyper.xml',\n",
       "  'description': 'SNP and indel caller',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/gatk2/gatk2_unified_genotyper/2.8.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fgatk2%2Fgatk2_unified_genotyper%2F2.8.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Unified Genotyper',\n",
       "  'panel_section_id': 'ngs:_gatk',\n",
       "  'panel_section_name': 'NGS: GATK',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '84584664264c',\n",
       "   'name': 'gatk2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.8.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/macs2/6d4babad010f/macs2/macs2_callpeak.xml',\n",
       "  'description': 'Call peaks from alignment results',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_callpeak/2.1.0.20151222.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fmacs2%2Fmacs2_callpeak%2F2.1.0.20151222.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MACS2 callpeak',\n",
       "  'panel_section_id': 'ngs:_peak_calling',\n",
       "  'panel_section_name': 'NGS: Peak Calling',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '6d4babad010f',\n",
       "   'name': 'macs2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.1.0.20151222.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/ulfschaefer/filter_vcf/cd59be4a7fe3/filter_vcf/filter_vcf.xml',\n",
       "  'description': 'filters a VCF file',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/ulfschaefer/filter_vcf/filter_vcf/1.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fulfschaefer%2Ffilter_vcf%2Ffilter_vcf%2F1.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Filter VCF',\n",
       "  'panel_section_id': 'ngs:_variant_analysis',\n",
       "  'panel_section_name': 'NGS: Variant Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'cd59be4a7fe3',\n",
       "   'name': 'filter_vcf',\n",
       "   'owner': 'ulfschaefer',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/288462ec2630/text_processing/find_and_replace.xml',\n",
       "  'description': 'parts of text',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_find_and_replace/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Ftext_processing%2Ftp_find_and_replace%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Replace',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '288462ec2630',\n",
       "   'name': 'text_processing',\n",
       "   'owner': 'bgruening',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/f8a9a5eaca8a/trimmomatic/trimmomatic.xml',\n",
       "  'description': 'flexible read trimming tool for Illumina NGS data',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/pjbriggs/trimmomatic/trimmomatic/0.32.3',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fpjbriggs%2Ftrimmomatic%2Ftrimmomatic%2F0.32.3',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Trimmomatic',\n",
       "  'panel_section_id': 'ngs:_qc_and_manipulation',\n",
       "  'panel_section_name': 'NGS: QC and manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8a9a5eaca8a',\n",
       "   'name': 'trimmomatic',\n",
       "   'owner': 'pjbriggs',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.32.3'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_revseq.xml',\n",
       "  'description': 'Reverse and complement a sequence',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: revseq82/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+revseq82%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'revseq',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/subtractBed.xml',\n",
       "  'description': 'remove intervals based on overlaps',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_subtractbed/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_subtractbed%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'SubtractBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/b723c120161a/sra_tools/fastq_dump.xml',\n",
       "  'description': 'in FASTQ/A format from NCBI SRA.',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/sra_tools/fastq_dump/1.2.5',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsra_tools%2Ffastq_dump%2F1.2.5',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Extract reads',\n",
       "  'panel_section_id': 'ncbi_sra_tools',\n",
       "  'panel_section_name': 'NCBI SRA Tools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b723c120161a',\n",
       "   'name': 'sra_tools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.2.5'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/288462ec2630/text_processing/sort_rows.xml',\n",
       "  'description': 'according to their columns',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sort_rows/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Ftext_processing%2Ftp_sort_rows%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Sort a row',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '288462ec2630',\n",
       "   'name': 'text_processing',\n",
       "   'owner': 'bgruening',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_chaos.xml',\n",
       "  'description': 'Create a chaos game representation plot for a sequence',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: chaos7/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+chaos7%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'chaos',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_textsearch.xml',\n",
       "  'description': 'Search sequence documentation. Slow, use SRS and Entrez!',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: textsearch98/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+textsearch98%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'textsearch',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/annotation_profiler/3b33da018e74/annotation_profiler/annotation_profiler.xml',\n",
       "  'description': 'for a set of genomic intervals',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/annotation_profiler/Annotation_Profiler_0/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fannotation_profiler%2FAnnotation_Profiler_0%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Profile Annotations',\n",
       "  'panel_section_id': 'operate_on_genomic_intervals',\n",
       "  'panel_section_name': 'Operate on Genomic Intervals',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '3b33da018e74',\n",
       "   'name': 'annotation_profiler',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/uniq.xml',\n",
       "  'description': 'occurrences of each record',\n",
       "  'id': 'Count1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=Count1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Count',\n",
       "  'panel_section_id': 'stats',\n",
       "  'panel_section_name': 'Statistics',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.2'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/288462ec2630/text_processing/sed.xml',\n",
       "  'description': 'with sed',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/bgruening/text_processing/tp_sed_tool/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fbgruening%2Ftext_processing%2Ftp_sed_tool%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Text transformation',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '288462ec2630',\n",
       "   'name': 'text_processing',\n",
       "   'owner': 'bgruening',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_primersearch.xml',\n",
       "  'description': 'Searches DNA sequences for matches with primer pairs',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: primersearch81/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+primersearch81%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'primersearch',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/abyss/f048033da666/abyss/abyss-pe.xml',\n",
       "  'description': 'de novo sequence assembler',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/abyss/abyss-pe/1.9.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fabyss%2Fabyss-pe%2F1.9.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'ABySS',\n",
       "  'panel_section_id': 'ngs:_de_novo',\n",
       "  'panel_section_name': 'NGS: De Novo',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f048033da666',\n",
       "   'name': 'abyss',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.9.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/bedtools/f8b7dc21b4ee/bedtools/jaccardBed.xml',\n",
       "  'description': 'calculate the distribution of relative distances between two files',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/bedtools/bedtools_jaccard/2.24.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fbedtools%2Fbedtools_jaccard%2F2.24.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'JaccardBed',\n",
       "  'panel_section_id': 'bedtools',\n",
       "  'panel_section_name': 'BEDTools',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'f8b7dc21b4ee',\n",
       "   'name': 'bedtools',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.24.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_wobble.xml',\n",
       "  'description': 'Wobble base plot',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: wobble108/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+wobble108%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'wobble',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/maf/maf_stats.xml',\n",
       "  'description': 'Alignment coverage information',\n",
       "  'id': 'maf_stats1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=maf_stats1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MAF Coverage Stats',\n",
       "  'panel_section_id': 'fetchAlign',\n",
       "  'panel_section_name': 'Fetch Alignments',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/gatk2/84584664264c/gatk2/haplotype_caller.xml',\n",
       "  'description': 'Call SNPs and indels simultaneously via local de-novo assembly of haplotypes in an active region',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/gatk2/gatk2_haplotype_caller/2.8.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fgatk2%2Fgatk2_haplotype_caller%2F2.8.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Haplotype Caller',\n",
       "  'panel_section_id': 'ngs:_gatk',\n",
       "  'panel_section_name': 'NGS: GATK',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '84584664264c',\n",
       "   'name': 'gatk2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.8.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/gff/gtf_filter_by_attribute_values_list.xml',\n",
       "  'description': '',\n",
       "  'id': 'gtf_filter_by_attribute_values_list',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=gtf_filter_by_attribute_values_list',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Filter GTF data by attribute values_list',\n",
       "  'panel_section_id': 'filter',\n",
       "  'panel_section_name': 'Filter and Sort',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '0.1'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/7f3c448e119b/ncbi_blast_plus/tools/ncbi_blast_plus/ncbi_makeblastdb.xml',\n",
       "  'description': 'Make BLAST database',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/ncbi_blast_plus/ncbi_makeblastdb/0.1.07',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fncbi_blast_plus%2Fncbi_makeblastdb%2F0.1.07',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'NCBI BLAST+ makeblastdb',\n",
       "  'panel_section_id': 'ngs:_mapping',\n",
       "  'panel_section_name': 'NGS: Mapping',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '7f3c448e119b',\n",
       "   'name': 'ncbi_blast_plus',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '0.1.07'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/cuffnorm/a26af05ac218/cuffnorm/cuffnorm_wrapper.xml',\n",
       "  'description': 'Create normalized expression levels',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/cuffnorm/cuffnorm/2.2.1.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fcuffnorm%2Fcuffnorm%2F2.2.1.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Cuffnorm',\n",
       "  'panel_section_id': 'ngs:_rna_analysis',\n",
       "  'panel_section_name': 'NGS: RNA Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'a26af05ac218',\n",
       "   'name': 'cuffnorm',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.2.1.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/catWrapper.xml',\n",
       "  'description': 'tail-to-head',\n",
       "  'id': 'cat1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=cat1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Concatenate datasets',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/eupathdb.xml',\n",
       "  'description': 'server',\n",
       "  'id': 'eupathdb',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=eupathdb',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'EuPathDB',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/snpeff/e09ce114d240/snpeff/snpEff_databases.xml',\n",
       "  'description': '',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/snpeff/snpEff_databases/4.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fsnpeff%2FsnpEff_databases%2F4.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'SnpEff Available Databases',\n",
       "  'panel_section_id': 'ngs:_variant_analysis',\n",
       "  'panel_section_name': 'NGS: Variant Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'e09ce114d240',\n",
       "   'name': 'snpeff',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '4.0.0'},\n",
       " {'config_file': '/shed_tools/testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/8bb5ea34809d/stacks/STACKS_prepare_population_map.xml',\n",
       "  'description': 'for STACKS denovomap and refmap',\n",
       "  'id': 'testtoolshed.g2.bx.psu.edu/repos/tiagoantao/stacks/STACKSpreparepopmap/1.2.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=testtoolshed.g2.bx.psu.edu%2Frepos%2Ftiagoantao%2Fstacks%2FSTACKSpreparepopmap%2F1.2.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'STACKS : Prepare population map file ',\n",
       "  'panel_section_id': 'ngs:_rads',\n",
       "  'panel_section_name': 'NGS: RADs',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '8bb5ea34809d',\n",
       "   'name': 'stacks',\n",
       "   'owner': 'tiagoantao',\n",
       "   'tool_shed': 'testtoolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.2.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_descseq.xml',\n",
       "  'description': 'Alter the name or description of a sequence',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: descseq21/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+descseq21%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'descseq',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_RevertOriginalBaseQualitiesAndAddMateCigar.xml',\n",
       "  'description': 'revert the original base qualities and add the mate cigar tag',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_RevertOriginalBaseQualitiesAndAddMateCigar/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_RevertOriginalBaseQualitiesAndAddMateCigar%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'RevertOriginalBaseQualitiesAndAddMateCigar',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_cirdna.xml',\n",
       "  'description': 'Draws circular maps of DNA constructs',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: cirdna11/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+cirdna11%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'cirdna',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/50f204c82159/taxonomy_krona_chart/taxonomy_krona_chart.xml',\n",
       "  'description': 'from taxonomic profile',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/crs4/taxonomy_krona_chart/taxonomy_krona_chart/2.6.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fcrs4%2Ftaxonomy_krona_chart%2Ftaxonomy_krona_chart%2F2.6.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Krona pie chart',\n",
       "  'panel_section_id': 'metagenomic_analysis',\n",
       "  'panel_section_name': 'Metagenomic Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '50f204c82159',\n",
       "   'name': 'taxonomy_krona_chart',\n",
       "   'owner': 'crs4',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.6.0'},\n",
       " {'config_file': '/galaxy-central/tools/filters/bed2gff.xml',\n",
       "  'description': 'converter',\n",
       "  'id': 'bed2gff1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=bed2gff1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'BED-to-GFF',\n",
       "  'panel_section_id': 'convert',\n",
       "  'panel_section_name': 'Convert Formats',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '2.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/yeastmine.xml',\n",
       "  'description': 'server',\n",
       "  'id': 'yeastmine',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=yeastmine',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'YeastMine',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/stats/filtering.xml',\n",
       "  'description': 'data on any column using simple expressions',\n",
       "  'id': 'Filter1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=Filter1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Filter',\n",
       "  'panel_section_id': 'filter',\n",
       "  'panel_section_name': 'Filter and Sort',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.1.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/macs2/6d4babad010f/macs2/macs2_bdgcmp.xml',\n",
       "  'description': 'Deduct noise by comparing two signal tracks in bedGraph',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/macs2/macs2_bdgcmp/2.1.0.20151222.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fmacs2%2Fmacs2_bdgcmp%2F2.1.0.20151222.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'MACS2 bdgcmp',\n",
       "  'panel_section_id': 'ngs:_peak_calling',\n",
       "  'panel_section_name': 'NGS: Peak Calling',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '6d4babad010f',\n",
       "   'name': 'macs2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.1.0.20151222.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_isochore.xml',\n",
       "  'description': 'Plots isochores in large DNA sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: isochore47/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+isochore47%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'isochore',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/data_source/wormbase_test.xml',\n",
       "  'description': 'test server',\n",
       "  'id': 'wormbase_test',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner/data_source_redirect?tool_id=wormbase_test',\n",
       "  'min_width': '800',\n",
       "  'model_class': 'DataSourceTool',\n",
       "  'name': 'Wormbase',\n",
       "  'panel_section_id': 'getext',\n",
       "  'panel_section_name': 'Get Data',\n",
       "  'target': '_top',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/lib/galaxy/datatypes/converters/picard_interval_list_to_bed6_converter.xml',\n",
       "  'description': 'converter',\n",
       "  'id': 'CONVERTER_picard_interval_list_to_bed6',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=CONVERTER_picard_interval_list_to_bed6',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Convert Picard Interval List to BED6',\n",
       "  'panel_section_id': None,\n",
       "  'panel_section_name': None,\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/galaxy-central/tools/extract/extract_genomic_dna.xml',\n",
       "  'description': 'using coordinates from assembled/unassembled genomes',\n",
       "  'id': 'Extract genomic DNA 1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=Extract+genomic+DNA+1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Extract Genomic DNA',\n",
       "  'panel_section_id': 'fetchSeq',\n",
       "  'panel_section_name': 'Fetch Sequences',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '2.2.3'},\n",
       " {'config_file': '/galaxy-central/tools/filters/remove_beginning.xml',\n",
       "  'description': 'of a file',\n",
       "  'id': 'Remove beginning1',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=Remove+beginning1',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Remove beginning',\n",
       "  'panel_section_id': 'textutil',\n",
       "  'panel_section_name': 'Text Manipulation',\n",
       "  'target': 'galaxy_main',\n",
       "  'version': '1.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/picard/efc56ee1ade4/picard/picard_AddOrReplaceReadGroups.xml',\n",
       "  'description': 'add or replaces read group information',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/picard/picard_AddOrReplaceReadGroups/1.136.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fpicard%2Fpicard_AddOrReplaceReadGroups%2F1.136.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'AddOrReplaceReadGroups',\n",
       "  'panel_section_id': 'ngs:_picard',\n",
       "  'panel_section_name': 'NGS: Picard',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'efc56ee1ade4',\n",
       "   'name': 'picard',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.136.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_wordmatch.xml',\n",
       "  'description': 'Finds all exact matches of a given size between 2 sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: wordmatch110/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+wordmatch110%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'wordmatch',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_backtranseq.xml',\n",
       "  'description': 'Back translate a protein sequence',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: backtranseq2/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+backtranseq2%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'backtranseq',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/b810c96613ee/emboss_5/emboss_5/emboss_dotmatcher.xml',\n",
       "  'description': 'Displays a thresholded dotplot of two sequences',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/devteam/emboss_5/EMBOSS: dotmatcher24/5.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Femboss_5%2FEMBOSS%3A+dotmatcher24%2F5.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'dotmatcher',\n",
       "  'panel_section_id': 'emboss',\n",
       "  'panel_section_name': 'EMBOSS',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': 'b810c96613ee',\n",
       "   'name': 'emboss_5',\n",
       "   'owner': 'devteam',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '5.0.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/iuc/gatk2/84584664264c/gatk2/variant_select.xml',\n",
       "  'description': 'from VCF files',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/iuc/gatk2/gatk2_variant_select/2.8.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fiuc%2Fgatk2%2Fgatk2_variant_select%2F2.8.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Select Variants',\n",
       "  'panel_section_id': 'ngs:_gatk',\n",
       "  'panel_section_name': 'NGS: GATK',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '84584664264c',\n",
       "   'name': 'gatk2',\n",
       "   'owner': 'iuc',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '2.8.0'},\n",
       " {'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/48b5f719e36a/str_fm/microsatpurity.xml',\n",
       "  'description': 'of a specific column',\n",
       "  'id': 'toolshed.g2.bx.psu.edu/repos/arkarachai-fungtammasan/str_fm/microsatpurity/1.0.0',\n",
       "  'labels': [],\n",
       "  'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Farkarachai-fungtammasan%2Fstr_fm%2Fmicrosatpurity%2F1.0.0',\n",
       "  'min_width': -1,\n",
       "  'model_class': 'Tool',\n",
       "  'name': 'Select uninterrupted STRs',\n",
       "  'panel_section_id': 'str-fm:_microsatellite_analysis',\n",
       "  'panel_section_name': 'STR-FM: Microsatellite Analysis',\n",
       "  'target': 'galaxy_main',\n",
       "  'tool_shed_repository': {'changeset_revision': '48b5f719e36a',\n",
       "   'name': 'str_fm',\n",
       "   'owner': 'arkarachai-fungtammasan',\n",
       "   'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       "  'version': '1.0.0'}]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gi.tools.get_tools()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'config_file': '/shed_tools/toolshed.g2.bx.psu.edu/repos/devteam/vcf2tsv/d6c944f1581a/vcf2tsv/vcf2tsv.xml',\n",
       " 'description': 'Convert VCF data into TAB-delimited format',\n",
       " 'id': 'toolshed.g2.bx.psu.edu/repos/devteam/vcf2tsv/vcf2tsv/0.0.3',\n",
       " 'inputs': [{'argument': None,\n",
       "   'edam_formats': ['format_3016'],\n",
       "   'extensions': ['vcf'],\n",
       "   'help': '',\n",
       "   'hidden': False,\n",
       "   'is_dynamic': False,\n",
       "   'label': 'Select VCF dataset to convert',\n",
       "   'model_class': 'DataToolParameter',\n",
       "   'multiple': False,\n",
       "   'name': 'input',\n",
       "   'optional': False,\n",
       "   'options': {'hda': [], 'hdca': []},\n",
       "   'type': 'data'},\n",
       "  {'argument': None,\n",
       "   'falsevalue': '',\n",
       "   'help': '-g option',\n",
       "   'hidden': False,\n",
       "   'is_dynamic': False,\n",
       "   'label': 'Report data per sample',\n",
       "   'model_class': 'BooleanToolParameter',\n",
       "   'name': 'g_option',\n",
       "   'optional': False,\n",
       "   'truevalue': '-g',\n",
       "   'type': 'boolean',\n",
       "   'value': True},\n",
       "  {'argument': None,\n",
       "   'display': None,\n",
       "   'help': '-n option',\n",
       "   'hidden': False,\n",
       "   'is_dynamic': False,\n",
       "   'label': 'Fill empty fields with',\n",
       "   'model_class': 'SelectToolParameter',\n",
       "   'multiple': False,\n",
       "   'name': 'null_filler',\n",
       "   'optional': False,\n",
       "   'options': [['Nothing', '', False],\n",
       "    ['.', '.', False],\n",
       "    ['*', '*', False],\n",
       "    ['NULL', 'NULL', False]],\n",
       "   'type': 'select',\n",
       "   'value': ''}],\n",
       " 'labels': [],\n",
       " 'link': '/tool_runner?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Fvcf2tsv%2Fvcf2tsv%2F0.0.3',\n",
       " 'min_width': -1,\n",
       " 'model_class': 'Tool',\n",
       " 'name': 'VCFtoTab-delimited:',\n",
       " 'outputs': [{'edam_format': 'format_3475',\n",
       "   'format': 'tabular',\n",
       "   'hidden': False,\n",
       "   'label': '',\n",
       "   'model_class': 'ToolOutput',\n",
       "   'name': 'out_file1'}],\n",
       " 'panel_section_id': 'ngs:_vcf_manipulation',\n",
       " 'panel_section_name': 'NGS: VCF Manipulation',\n",
       " 'target': 'galaxy_main',\n",
       " 'tool_shed_repository': {'changeset_revision': 'd6c944f1581a',\n",
       "  'name': 'vcf2tsv',\n",
       "  'owner': 'devteam',\n",
       "  'tool_shed': 'toolshed.g2.bx.psu.edu'},\n",
       " 'version': '0.0.3'}"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vcf2tsv = gi.tools.get_tools(name='VCFtoTab-delimited:')[0]\n",
    "gi.tools.show_tool(vcf2tsv['id'], io_details=True, link_details=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'annotation': None,\n",
       " 'contents_url': '/api/histories/7d83dea28f2ec7cc/contents',\n",
       " 'create_time': '2016-08-10T17:12:34.918207',\n",
       " 'deleted': False,\n",
       " 'empty': True,\n",
       " 'genome_build': None,\n",
       " 'id': '7d83dea28f2ec7cc',\n",
       " 'importable': False,\n",
       " 'model_class': 'History',\n",
       " 'name': 'data_science_example',\n",
       " 'published': False,\n",
       " 'purged': False,\n",
       " 'size': 0,\n",
       " 'slug': None,\n",
       " 'state': 'new',\n",
       " 'state_details': {'discarded': 0,\n",
       "  'empty': 0,\n",
       "  'error': 0,\n",
       "  'failed_metadata': 0,\n",
       "  'new': 0,\n",
       "  'ok': 0,\n",
       "  'paused': 0,\n",
       "  'queued': 0,\n",
       "  'running': 0,\n",
       "  'setting_metadata': 0,\n",
       "  'upload': 0},\n",
       " 'state_ids': {'discarded': [],\n",
       "  'empty': [],\n",
       "  'error': [],\n",
       "  'failed_metadata': [],\n",
       "  'new': [],\n",
       "  'ok': [],\n",
       "  'paused': [],\n",
       "  'queued': [],\n",
       "  'running': [],\n",
       "  'setting_metadata': [],\n",
       "  'upload': []},\n",
       " 'tags': [],\n",
       " 'update_time': '2016-08-10T17:12:34.918225',\n",
       " 'url': '/api/histories/7d83dea28f2ec7cc',\n",
       " 'user_id': 'f2db41e1fa331b3e',\n",
       " 'username_and_slug': None}"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ds_history"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "def dataset_to_param(dataset):\n",
    "    return dict(src='hda', id=dataset['id'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'null_filter': '', 'input': {'id': '56573e25b995be9d', 'src': 'hda'}, 'g_option': True}\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'implicit_collections': [],\n",
       " 'jobs': [{'create_time': '2016-08-10T17:13:05.271059',\n",
       "   'exit_code': None,\n",
       "   'id': 'd49aa4a2fce5bb0a',\n",
       "   'model_class': 'Job',\n",
       "   'state': 'new',\n",
       "   'tool_id': 'toolshed.g2.bx.psu.edu/repos/devteam/vcf2tsv/vcf2tsv/0.0.3',\n",
       "   'update_time': '2016-08-10T17:13:05.271072'}],\n",
       " 'output_collections': [],\n",
       " 'outputs': [{'create_time': '2016-08-10T17:13:05.217759',\n",
       "   'data_type': 'galaxy.datatypes.tabular.Tabular',\n",
       "   'deleted': False,\n",
       "   'file_ext': 'tabular',\n",
       "   'file_size': 0,\n",
       "   'genome_build': '?',\n",
       "   'hda_ldda': 'hda',\n",
       "   'hid': 2,\n",
       "   'history_content_type': 'dataset',\n",
       "   'history_id': '7d83dea28f2ec7cc',\n",
       "   'id': '2700ea79f26a88c2',\n",
       "   'metadata_column_names': None,\n",
       "   'metadata_column_types': None,\n",
       "   'metadata_columns': None,\n",
       "   'metadata_comment_lines': None,\n",
       "   'metadata_data_lines': None,\n",
       "   'metadata_dbkey': '?',\n",
       "   'metadata_delimiter': '\\t',\n",
       "   'misc_blurb': 'queued',\n",
       "   'misc_info': None,\n",
       "   'model_class': 'HistoryDatasetAssociation',\n",
       "   'name': 'VCFtoTab-delimited: on data 1',\n",
       "   'output_name': 'out_file1',\n",
       "   'peek': '<table cellspacing=\"0\" cellpadding=\"3\"><tr></tr></table>',\n",
       "   'purged': False,\n",
       "   'state': 'new',\n",
       "   'tags': [],\n",
       "   'update_time': '2016-08-10T17:13:05.267781',\n",
       "   'uuid': '61e4adf2-7b0c-4c29-ba14-97bb15369ba0',\n",
       "   'visible': True}]}"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tool_inputs = {'input': dataset_to_param(vcf_ds), 'g_option': True, 'null_filter': ''}\n",
    "#hid!\n",
    "print(tool_inputs)\n",
    "gi.tools.run_tool(ds_history['id'], vcf2tsv['id'], tool_inputs=tool_inputs)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.5.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
